% Document-defining preamble
\documentclass[12pt,twoside,openright]{report} 
\usepackage[phd]{edmaths}
\usepackage[T1]{fontenc}

% Title page meta-data 
\title{
Statistical and Machine Learning Approaches to Genomic Medicine \vspace{100pt} \\ 
\includegraphics[width=3in]{figures/misc/edi_uni_logo.png} \\
\vspace{50pt}
}
\author{
Jacob R. Bradley
}
\date{2023}

% Sections
%\renewcommand{\thesection}{\arabic{section}} % removes chapter numbers

% Glossary
\usepackage[acronym, toc]{glossaries-extra} % LINDEN DON'TUNCOMMENT
\setabbreviationstyle[acronym]{long-short}
\glssetcategoryattribute{acronym}{nohyperfirst}{true}
\renewcommand*{\glsdonohyperlink}[2]{%
 {\glsxtrprotectlinks \glsdohypertarget{#1}{#2}}}
\usepackage[
% draft, 
colorlinks,linkcolor=black,citecolor=blue,urlcolor=blue,breaklinks = true
]{hyperref} %makes glossary elements black

\makeglossaries 
\newacronym{roc}{ROC}{receiver operating characteristic}
\newacronym{tcga}{TCGA}{The Cancer Genome Atlas}
\newacronym{pv}{PV}{polycythaemia vera}
\newacronym{tmb}{TMB}{tumour mutation burden}
\newacronym{icb}{ICB}{immune checkpoint blockade}
\newacronym{ici}{ICI}{immune checkpoint inhibitor}
\newacronym{msi}{MSI}{micro-satellite instability}
\newacronym{lasso}{LASSO}{least absolute shrinkage and selection operator}
\newacronym{tib}{TIB}{tumour indel burden}
\newacronym{ctla4}{CTLA-4}{cytotoxic T-lymphocyte associated protein 4}
\newacronym{pdl1}{PD-L1}{programmed death ligand 1}
\newacronym{wes}{WES}{whole-exome sequencing}
\newacronym{ctdna}{ctDNA}{circulating tumour DNA}
\newacronym{bmr}{BMR}{background mutation rate}
\newacronym{nsclc}{NSCLC}{non-small cell lung cancer}
\newacronym{auprc}{AUPRC}{area under precision-recall curve}
\newacronym{ectmb}{ecTMB}{estimation and classification of tumour mutation burden}
\newacronym{mcmc}{MCMC}{Markov chain Monte Carlo}
\newacronym{dna}{DNA}{deoxyribonucleic acid}
\newacronym{rna}{RNA}{ribonucleic acid}
\newacronym{cdna}{cDNA}{complementary DNA}
\newacronym{gdna}{gDNA}{genomic DNA}
\newacronym{qrtlamp}{qRT-LAMP}{quantitative reverse-transcription loop-mediated isothermal amplification}
\newacronym{lamp}{LAMP}{loop-mediated isothermal amplification}
\newacronym{qpcr}{qPCR}{quantitative polymerase chain reaction}
\newacronym{pcr}{PCR}{polymerase chain reaction}
\newacronym{mrna}{mRNA}{messenger RNA}
\newacronym{tt}{Tt}{time-to-threshold}
\newacronym{ns}{NS}{NanoString}
\newacronym{fip}{FIP}{forward inner primer}
\newacronym{bip}{BIP}{backward inner primer}
\newacronym{f3}{F3}{forward outer primer}
\newacronym{b3}{B3}{backward outer primer}
\newacronym{lf}{LF}{loop forward}
\newacronym{lb}{LB}{loop backward}
\newacronym{tm}{Tm}{melting temperature}
\newacronym{loopsis}{LOO-PSIS}{leave-one-out Pareto smoothed importance sampling}
\newacronym{ivt}{IVT}{\emph{in vitro}-transcribed}
\newacronym{hmc}{HMC}{Hamiltonian Monte Carlo}
\newacronym{wgs}{WGS}{whole-genome sequencing}
\newacronym{ipw}{IPW}{inverse-probability weighting}
\newacronym{cate}{CATE}{conditional average treatment effect}
\newacronym{uv}{UV}{ultraviolet}
\newacronym{os}{OS}{overall survival}
\newacronym{maf}{MAF}{mutation annotated format}
\newacronym{rdrp}{RdRp}{RNA-dependent RNA polymerase}
\newacronym{rrna}{rRNA}{ribosomal RNA}
\newacronym{ols}{OLS}{ordinary least squares}
\newacronym{ir}{IR}{image recognition}
\newacronym{nlp}{NLP}{natural language processing}
\newacronym[plural = SVMs, firstplural = support vector machines]{svm}{SVM}{support vector machine}
\newacronym{sarscov2}{SARS-CoV-2}{severe acute respiratory syndrome coronavirus 2}
\newacronym{pku}{PKU}{phenylketonuria}
\newacronym[plural = HTEs, firstplural = heterogeneous treatment effects]{hte}{HTE}{heterogeneous treatment effect}
\newacronym[plural = ATEs, firstplural = average treatment effects]{ate}{ATE}{average treatment effect}
\newacronym{pgm}{PGM}{probabilistic graphical model}
\newacronym{iqr}{IQR}{inter-quartile range}
\newacronym{mskcc}{MSKCC}{the memorial Sloan Kettering cancer centre}
\newacronym{pfs}{PFS}{progression-free survival}
\newacronym{rct}{RCT}{randomised controlled trial}
\newacronym[plural = ANNs, firstplural = artificial neural networks]{ann}{ANN}{artificial neural network}
\newacronym{vcf}{VCF}{variant called format}
\newacronym{ipm}{IPM}{integral probability metric}

% \newglossaryentry{latex}
% {
%     name=latex,
%     description={Is a markup language specially $\lambda$ suited 
%     for scientific documents}
% } % LINDEN UNCOMMENT

% \newglossaryentry{lambda}
% {
%     name=\lambda,
%     description={Is a markup language specially $\lambda$ suited 
%     for scientific documents}
% } % LINDEN UNCOMMENT

% Typesetting
\usepackage{multirow,latexsym,amsmath,amssymb,amsfonts,amsthm,bbm,mathrsfs,breakcites,dsfont,xcolor, enumerate, url, bm}

\theoremstyle{definition}
\newtheorem{definition}{Definition}[section]
\newtheorem{example}{Example}[section]
\newtheorem{technique}{Technique}[section]
\DeclareMathOperator*{\argmax}{arg\,max}
\DeclareMathOperator*{\argmin}{arg\,min}
\newcommand{\var}{\mathrm{Var}}
\newcommand{\cov}{\mathrm{Cov}}
\newcommand{\bbE}{\mathbb{E}}
\newcommand{\pihat}{\hat{\pi}}
\newcommand{\muhat}{\hat{\mu}}

% For adding quotes to beginning of sections
\usepackage{epigraph} 

% Typesetting code
\usepackage{listings}

% Editing tools (for when Tim gets involved)
\usepackage[draft,inline,nomargin,index]{fixme}
\fxsetup{theme=color,mode=multiuser}
\FXRegisterAuthor{tc}{atc}{\color{green} TC}
\FXRegisterAuthor{jb}{abc}{\color{red} JB}
\usepackage{outlines}

% Figures 
\usepackage{graphicx}
\graphicspath{{figures/}{../figures/}}
\usepackage{tikz-cd}
\usepackage[ruled,vlined]{algorithm2e}


% Bibliography and referencing
\usepackage{natbib}
% \usepackage{xr}
\usepackage{subfiles} % Manages subfiles
\setcitestyle{authoryear, open={(},close={)}}

\def\dobib{
    \bibliographystyle{natbib}
    \bibliography{references.bib} % Renders bibliographies when in single chapter view.
}

\begin{document}
\def\dobib{} % Disable chapter-specific bibliographies
\lstset{showstringspaces=false,language=R} % inline code settings

\maketitle

\declaration

\dedication{To Morton...}

\subfile{chapters/abstract}

\chapter*{Lay Summary}
\label{chap:laysummary}
\addcontentsline{toc}{chapter}{\nameref{chap:laysummary}}

\subfile{chapters/laysummary}

% \printglossary % LINDEN UNCOMMENT
\printglossary[type=\acronymtype]

\tableofcontents
~\\
{\color{red} FOR TIM:
I'm going to keep adding to this list of sections that are broadly speaking ready to be read and for comments. If you find you've run out of stuff to read then just pop me a message and I'll point you to stuff that is essentially done but needs more attention.
\begin{outline}
    \1 Abstract. This should now be locked in, as we've included in when reaching out to examiners and in filing NITS.
    \1 Lay summary. I've tried to make this fairly fun and breezy. I'm guessing it's not the most important section in the world so have been pretty free with it, but let me know if it's not hitting the right tone.
    \1 Section 1.1. This is a general overview of the type of content in the entire thesis (but not discussing any specifics of layout).
        \2 1.1.0 Very general motivating questions for problems looked at across the thesis, and outline of the introduction chapter.
        \2 1.1.1 Talking about what new types of genomics data are available and what problems in medicine they've been used to solve. First airtime for ideas like `subtyping' (motivation for Chapter 2) and `biomarkers to accompany therapy` (motivation for Chapter 3). Finish up with a mention of the difficulties combining disparate datasets (motivation for Chapter 4), and practical problems with the data (motivate the need for statistics and lead in to 1.1.2)
        \2 1.1.2 Overview of statistical learning and machine learning. Bit of loose definition work talking about `prediction', `regression', `model fitting'. While ML is in the title of the thesis for cynical professional reasons, motivate why `can't we just use fancy ML stuff' isn't always/often an option with the data we're working with.
    \1 Section 1.2 Biologial background necessary to understand what's going to happen in the rest of the thesis.
        \2 1.2.1 The core relationships between genomics quantities (DNA, RNA, protein). Should give enough detail that when data types/specific datasets are introduced later on people will know where they fit into the bigger biology picture.
        \2 1.2.2 Motivating that molecular biology as a whole has been useful in medicine, with a particular focus on how its impacted cancer research.
        \2 1.2.3 Overview of common technologies in genomics. Should be enough that readers know what we're talking about in terms of targeted sequencing in Chapters 3 and 4, and to get enough of a grasp of PCR to understand what LAMP is trying to do and why it's different in Chapter 2.
    \1 Section 2. This is obviously an entire chapter that will be fresh-ish for you. It's fairly well developed (hopefully). I don't see realistically making huge changes to the content, but it'd be great to get a sense of how undertandable it is with the introduction, and any more context that needs giving.
    \1 Section 4. This is all getting thrown together very fast, and I think should be the next thing you read after finished intro sections and Chapter~2. At the moment it's basically got results for half of the methods I want to demo, with the other methods following a similar trajectory. 
        \2 Sections 4.1-4.3. These outline all aspects of heterogeneous treatment effect estimation up to (but not yet including) survival outcomes. There's a reasonable amount of writing there but it'll need editing.
        \2 4.5 This should now lay out the experiments I'm hoping to have in place for causal survival forests and regularisation based HTE estimation, with only causal survival forests so far in place.
    % \1 Appendices. I think these are low priorities. Appendices A and B are for LAMP, Appendix C for TMB. I haven't fleshed them out, and think this might actually be a good task for the week after you leave in July. 
    %     \2 Appendix A: We have an R package for feature generation for LAMP, and I also want to talk about clever codey workflow management stuff I've done. 
    %     \2 Appendix B: Summary figures for Chapter 2 that don't make it into the main text. 
    %     \2 Appendix C: This dicusses how the \texttt{ICBioMark} package is used. I did experiment with putting it in its own section in the chapter, but somehow it felt more at home in methods. Eventually after I decided there was enough to say about LAMP code to have an appendix, I decided to give it its own. Let me know what you think.
    
\end{outline}
}

\chapter{Introduction \label{chap:intro}}

\subfile{chapters/chapter1}


\chapter{Bayesian analysis for optimising the LAMP gene expression assay \label{chap:lamp_modelling}}

\subfile{chapters/chapter2}

\chapter{Data-driven design of targeted gene panels for estimating immunotherapy biomarkers \label{chap:tmb_estimation}}

\subfile{chapters/chapter3}

\chapter{Causal survival analysis in oncology\label{chap:causal_genomics}}

\subfile{chapters/chapter4}

\addcontentsline{toc}{chapter}{Bibliography}
\bibliographystyle{natbib}
\bibliography{references.bib}

\appendix

\chapter{Computational workflow for LAMP assay analysis}
\subfile{chapters/appendix1}

\chapter{Extened model summaries for LAMP assay analysis}
\subfile{chapters/appendix2}

\chapter{Open-source software accompanying TMB estimation}
\subfile{chapters/appendix3}

\end{document}
