% Document-defining preamble
\documentclass[12pt,twoside,openright]{report} 
\usepackage[phd]{edmaths}
\usepackage[T1]{fontenc}

% Title page meta-data 
\title{
Statistical and Machine Learning Approaches to Genomic Medicine \vspace{100pt} \\ 
\includegraphics[width=3in]{figures/misc/edi_uni_logo.png} \\
\vspace{50pt}
}
\author{
Jacob R. Bradley
}
\date{2023}

% Sections
%\renewcommand{\thesection}{\arabic{section}} % removes chapter numbers

% Glossary
\usepackage[acronym, toc]{glossaries-extra} % LINDEN DON'TUNCOMMENT
\setabbreviationstyle[acronym]{long-short}
\glssetcategoryattribute{acronym}{nohyperfirst}{true}
\renewcommand*{\glsdonohyperlink}[2]{%
 {\glsxtrprotectlinks \glsdohypertarget{#1}{#2}}}
\usepackage[
% draft, 
colorlinks,linkcolor=black,citecolor=blue,urlcolor=blue,breaklinks = true
]{hyperref} %makes glossary elements black

\makeglossaries 
\newacronym{roc}{ROC}{receiver operating characteristic}
\newacronym{tcga}{TCGA}{The Cancer Genome Atlas}
\newacronym{pv}{PV}{polycythaemia vera}
\newacronym{tmb}{TMB}{tumour mutation burden}
\newacronym{icb}{ICB}{immune checkpoint blockade}
\newacronym{ici}{ICI}{immune checkpoint inhibitor}
\newacronym{msi}{MSI}{micro-satellite instability}
\newacronym{lasso}{LASSO}{least absolute shrinkage and selection operator}
\newacronym{tib}{TIB}{tumour indel burden}
\newacronym{ctla4}{CTLA-4}{cytotoxic T-lymphocyte associated protein 4}
\newacronym{pdl1}{PD-L1}{programmed death ligand 1}
\newacronym{wes}{WES}{whole-exome sequencing}
\newacronym{ctdna}{ctDNA}{circulating tumour DNA}
\newacronym{bmr}{BMR}{background mutation rate}
\newacronym{nsclc}{NSCLC}{non-small cell lung cancer}
\newacronym{auprc}{AUPRC}{area under precision-recall curve}
\newacronym{ectmb}{ecTMB}{estimation and classification of tumour mutation burden}
\newacronym{mcmc}{MCMC}{Markov chain Monte Carlo}
\newacronym{dna}{DNA}{deoxyribonucleic acid}
\newacronym{rna}{RNA}{ribonucleic acid}
\newacronym{cdna}{cDNA}{complementary DNA}
\newacronym{gdna}{gDNA}{genomic DNA}
\newacronym{qrtlamp}{qRT-LAMP}{quantitative reverse-transcription loop-mediated isothermal amplification}
\newacronym{lamp}{LAMP}{loop-mediated isothermal amplification}
\newacronym{qpcr}{qPCR}{quantitative polymerase chain reaction}
\newacronym{pcr}{PCR}{polymerase chain reaction}
\newacronym{mrna}{mRNA}{messenger RNA}
\newacronym{tt}{Tt}{time-to-threshold}
\newacronym{ns}{NS}{NanoString}
\newacronym{fip}{FIP}{forward inner primer}
\newacronym{bip}{BIP}{backward inner primer}
\newacronym{f3}{F3}{forward outer primer}
\newacronym{b3}{B3}{backward outer primer}
\newacronym{lf}{LF}{loop forward}
\newacronym{lb}{LB}{loop backward}
\newacronym{tm}{Tm}{melting temperature}
\newacronym{loopsis}{LOO-PSIS}{leave-one-out Pareto smoothed importance sampling}
\newacronym{ivt}{IVT}{\emph{in vitro}-transcribed}
\newacronym{hmc}{HMC}{Hamiltonian Monte Carlo}
\newacronym{wgs}{WGS}{whole-genome sequencing}
\newacronym{ipw}{IPW}{inverse-probability weighting}
\newacronym{cate}{CATE}{conditional average treatment effect}
\newacronym{uv}{UV}{ultraviolet}
\newacronym{os}{OS}{overall survival}
\newacronym{maf}{MAF}{mutation annotated format}
\newacronym{rdrp}{RdRp}{RNA-dependent RNA polymerase}
\newacronym{rrna}{rRNA}{ribosomal RNA}
\newacronym{ols}{OLS}{ordinary least squares}
\newacronym{ir}{IR}{image recognition}
\newacronym{nlp}{NLP}{natural language processing}
\newacronym[plural = SVMs, firstplural = support vector machines]{svm}{SVM}{support vector machine}
\newacronym{sarscov2}{SARS-CoV-2}{severe acute respiratory syndrome coronavirus 2}
\newacronym{pku}{PKU}{phenylketonuria}
\newacronym[plural = HTEs, firstplural = heterogeneous treatment effects]{hte}{HTE}{heterogeneous treatment effect}
\newacronym[plural = ATEs, firstplural = average treatment effects]{ate}{ATE}{average treatment effect}
\newacronym{pgm}{PGM}{probabilistic graphical model}
\newacronym{iqr}{IQR}{inter-quartile range}
\newacronym{mskcc}{MSKCC}{the memorial Sloan Kettering cancer centre}
\newacronym{pfs}{PFS}{progression-free survival}
\newacronym{rct}{RCT}{randomised controlled trial}
\newacronym[plural = ANNs, firstplural = artificial neural networks]{ann}{ANN}{artificial neural network}
\newacronym{vcf}{VCF}{variant called format}
\newacronym{ipm}{IPM}{integral probability metric}
\newacronym[plural = CSFs, firstplural = causal survival forests]{csf}{CSF}{causal survival forest}
\newacronym{csa}{CSA}{counterfactual survival analysis}
\newacronym{rmst}{RMST}{restricted mean survival time}
\newacronym{ipcw}{IPCW}{inverse probability of censoring weighting}


% \newglossaryentry{latex}
% {
%     name=latex,
%     description={Is a markup language specially $\lambda$ suited 
%     for scientific documents}
% } % LINDEN UNCOMMENT

% \newglossaryentry{lambda}
% {
%     name=\lambda,
%     description={Is a markup language specially $\lambda$ suited 
%     for scientific documents}
% } % LINDEN UNCOMMENT

% Typesetting
\usepackage{multirow,latexsym,amsmath,amssymb,amsfonts,amsthm,bbm,mathrsfs,breakcites,dsfont,xcolor, enumerate, url, bm}

\theoremstyle{definition}
\newtheorem{definition}{Definition}[section]
\newtheorem{example}{Example}[section]
\newtheorem{technique}{Technique}[section]
\DeclareMathOperator*{\argmax}{arg\,max}
\DeclareMathOperator*{\argmin}{arg\,min}
\newcommand{\var}{\mathrm{Var}}
\newcommand{\cov}{\mathrm{Cov}}
\newcommand{\bbE}{\mathbb{E}}
\newcommand{\pihat}{\hat{\pi}}
\newcommand{\muhat}{\hat{\mu}}

% For adding quotes to beginning of sections
\usepackage{epigraph} 

% Typesetting code
\usepackage{listings}

% Editing tools (for when Tim gets involved)
\usepackage[draft,inline,nomargin,index]{fixme}
\fxsetup{theme=color,mode=multiuser}
\FXRegisterAuthor{tc}{atc}{\color{green} TC}
\FXRegisterAuthor{jb}{abc}{\color{red} JB}
\usepackage{outlines}

% Figures 
\usepackage{graphicx}
\graphicspath{{figures/}{../figures/}}
\usepackage{tikz-cd}
\usepackage[ruled,vlined]{algorithm2e}


% Bibliography and referencing
\usepackage{natbib}
% \usepackage{xr}
\usepackage{subfiles} % Manages subfiles
\setcitestyle{authoryear, open={(},close={)}}

\def\dobib{
    \bibliographystyle{natbib}
    \bibliography{references.bib} % Renders bibliographies when in single chapter view.
}

\begin{document}
\def\dobib{} % Disable chapter-specific bibliographies
\lstset{showstringspaces=false,language=R} % inline code settings

\maketitle

\declaration

\dedication{To Morton...}

\subfile{chapters/abstract}

\chapter*{Lay Summary}
\label{chap:laysummary}
\addcontentsline{toc}{chapter}{\nameref{chap:laysummary}}

\subfile{chapters/laysummary}

% \printglossary % LINDEN UNCOMMENT
\printglossary[type=\acronymtype]

\tableofcontents
~\\
{\color{red} FOR TIM:
Summary of progress.
\begin{outline}
    \1 Abstract. This should now be locked in, as we've included in when reaching out to examiners and in filing NITS.
    \1 Lay summary. I've tried to make this fairly fun and breezy. I'm guessing it's not the most important section in the world so have been pretty free with it, but let me know if it's not hitting the right tone.
    \1 Section 1. Sections 1.0-1.2 complete (with some revisions left to integrate), sections 1.3-1.4 now current focus.
    \1 Section 2. Most comments are now assimilated, but still need to standardise the final models that get used.
    \1 Section 3. Our paper. Not envisioning adding anything, code stuff now an appendix. Not sure I'll have time to make major changes.
    \1 Section 4. This should now be ready to read through! (Some finishing off to do in conclusions).
    \1 Appendices. Low priority. Am going to write while you're away (/potentially while I'm away).
    
    
\end{outline}
}

\chapter{Introduction \label{chap:intro}}
\subfile{chapters/chapter1}


\chapter{Bayesian analysis for optimising LAMP gene expression assays \label{chap:lamp_modelling}}
\subfile{chapters/chapter2}

\chapter{Data-driven design of targeted gene panels for estimating immunotherapy biomarkers \label{chap:tmb_estimation}}
\subfile{chapters/chapter3}

\chapter{Causal survival analysis in oncology\label{chap:causal_genomics}}
\subfile{chapters/chapter4}

\addcontentsline{toc}{chapter}{Bibliography}
\bibliographystyle{natbib}
\bibliography{references.bib}

\appendix

\chapter{Computational workflow for LAMP assay analysis}
\subfile{chapters/appendix1}

\chapter{Extened model summaries for LAMP assay analysis}
\subfile{chapters/appendix2}

\chapter{Open-source software accompanying TMB estimation}
\subfile{chapters/appendix3}

\end{document}
