% Document-defining preamble
\documentclass[12pt,twoside,openright]{report} 
\usepackage[phd]{edmaths}
\usepackage[T1]{fontenc}

% Title page meta-data 
\title{Statistical and Machine Learning Approaches to Genomic Medicine}
\author{Jacob R. Bradley}
\date{2023}

% Sections
%\renewcommand{\thesection}{\arabic{section}} % removes chapter numbers

% Glossary
\usepackage[acronym, toc]{glossaries-extra} % LINDEN DON'TUNCOMMENT
\setabbreviationstyle[acronym]{long-short}
\glssetcategoryattribute{acronym}{nohyperfirst}{true}
\renewcommand*{\glsdonohyperlink}[2]{%
 {\glsxtrprotectlinks \glsdohypertarget{#1}{#2}}}
\usepackage[
% draft, 
colorlinks,linkcolor=black,citecolor=blue,urlcolor=blue,breaklinks = true
]{hyperref} %makes glossary elements black

\makeglossaries 
\newacronym{roc}{ROC}{receiver operating characteristic}
\newacronym{tcga}{TCGA}{The Cancer Genome Atlas}
\newacronym{pv}{PV}{polycythaemia vera}
\newacronym{tmb}{TMB}{tumour mutation burden}
\newacronym{icb}{ICB}{immune checkpoint blockade}
\newacronym{ici}{ICI}{immune checkpoint inhibitor}
\newacronym{msi}{MSI}{micro-satellite instability}
\newacronym{lasso}{LASSO}{least absolute shrinkage and selection pperator}
\newacronym{tib}{TIB}{tumour indel burden}
\newacronym{ctla4}{CTLA-4}{cytotoxic T-lymphocyte associated protein 4}
\newacronym{pdl1}{PD-L1}{programmed death ligand 1}
\newacronym{wes}{WES}{whole-exome sequencing}
\newacronym{ctdna}{ctDNA}{circulating tumour DNA}
\newacronym{bmr}{BMR}{background mutation rate}
\newacronym{nsclc}{NSCLC}{non-small cell lung cancer}
\newacronym{auprc}{AUPRC}{area under precision-recall curve}
\newacronym{ectmb}{ecTMB}{estimation and classification of tumour mutation burden}
\newacronym{mcmc}{MCMC}{Markov chain Monte Carlo}
\newacronym{dna}{DNA}{deoxyribonucleic acid}
\newacronym{rna}{RNA}{ribonucleic acid}
\newacronym{cdna}{cDNA}{complementary DNA}
\newacronym{gdna}{gDNA}{genomic DNA}
\newacronym{qrtlamp}{qRT-LAMP}{quantitative reverse-transcription loop-mediated isothermal amplification}
\newacronym{lamp}{LAMP}{loop-mediated isothermal amplification}
\newacronym{qpcr}{qPCR}{quantitative polymerase chain reaction}
\newacronym{pcr}{PCR}{polymerase chain reaction}
\newacronym{mrna}{mRNA}{messenger RNA}
\newacronym{tt}{Tt}{time-to-threshold}
\newacronym{ns}{NS}{NanoString}
\newacronym{fip}{FIP}{forward inner primer}
\newacronym{bip}{BIP}{backward inner primer}
\newacronym{f3}{F3}{forward outer primer}
\newacronym{b3}{B3}{backward outer primer}
\newacronym{lf}{LF}{loop forward}
\newacronym{lb}{LB}{loop backward}
\newacronym{tm}{Tm}{melting temperature}
\newacronym{loopsis}{LOO-PSIS}{leave-one-out Pareto smoothed importance sampling}
\newacronym{ivt}{IVT}{\emph{in vitro}-transcribed}
\newacronym{hmc}{HMC}{Hamiltonian Monte Carlo}
\newacronym{wgs}{WGS}{whole-genome sequencing}
\newacronym{ipw}{IPW}{inverse-probability weighting}
\newacronym{cate}{CATE}{conditional average treatment effect}
\newacronym{uv}{UV}{ultraviolet}
\newacronym{os}{OS}{overall survival}
\newacronym{maf}{MAF}{mutation annotated format}
\newacronym{rdrp}{RdRp}{RNA-dependent RNA polymerase}
\newacronym{rrna}{rRNA}{ribosomal RNA}
\newacronym{ols}{OLS}{ordinary least squares}

% \newglossaryentry{latex}
% {
%     name=latex,
%     description={Is a markup language specially $\lambda$ suited 
%     for scientific documents}
% } % LINDEN UNCOMMENT

% \newglossaryentry{lambda}
% {
%     name=\lambda,
%     description={Is a markup language specially $\lambda$ suited 
%     for scientific documents}
% } % LINDEN UNCOMMENT

% Typesetting
\usepackage{multirow,latexsym,amsmath,amssymb,amsfonts,amsthm,bbm,mathrsfs,breakcites,dsfont,xcolor, enumerate, url, bm}

\theoremstyle{definition}
\newtheorem{definition}{Definition}[section]
\newtheorem{example}{Example}[section]
\newtheorem{technique}{Technique}[section]
\DeclareMathOperator*{\argmax}{arg\,max}
\DeclareMathOperator*{\argmin}{arg\,min}
\newcommand{\var}{\mathrm{Var}}
\newcommand{\cov}{\mathrm{Cov}}
\newcommand{\bbE}{\mathbb{E}}

% For adding quotes to beginning of sections
\usepackage{epigraph} 

% Typesetting code
\usepackage{listings}

% Editing tools (for when Tim gets involved)
\usepackage[draft,inline,nomargin,index]{fixme}
\fxsetup{theme=color,mode=multiuser}
\FXRegisterAuthor{tc}{atc}{\color{green} TC}
\FXRegisterAuthor{jb}{abc}{\color{red} JB}

% Figures 
\usepackage{graphicx}
\graphicspath{{figures/}{../figures/}}
\usepackage{tikz-cd}
\usepackage[ruled,vlined]{algorithm2e}


% Bibliography and referencing
\usepackage{natbib}
% \usepackage{xr}
\usepackage{subfiles} % Manages subfiles
\setcitestyle{authoryear, open={(},close={)}}

\def\dobib{
    \bibliographystyle{natbib}
    \bibliography{references.bib} % Renders bibliographies when in single chapter view.
}

\begin{document}
\def\dobib{} % Disable chapter-specific bibliographies
\lstset{showstringspaces=false,language=R} % inline code settings

\maketitle

\declaration

\dedication{To Morton...}

\subfile{chapters/abstract}

\chapter*{Lay Summary}
\label{chap:laysummary}
\addcontentsline{toc}{chapter}{\nameref{chap:laysummary}}

\subfile{chapters/laysummary}

% \printglossary % LINDEN UNCOMMENT
\printglossary[type=\acronymtype]

\tableofcontents

\chapter{Introduction \label{chap:intro}}

\subfile{chapters/chapter1}


\chapter{Hierarchical Bayesian modeling identifies key considerations in the
development of quantitative loop-mediated isothermal amplification
assays \label{chap:lamp_modelling}}

\subfile{chapters/chapter2}

\chapter{Data-driven design of targeted gene panels for estimating immunotherapy biomarkers \label{chap:tmb_estimation}}

\subfile{chapters/chapter3}

\chapter{Causal prediction of survival after immunotherapy with genomic mutation signatures \label{chap:causal_genomics}}

\subfile{chapters/chapter4}

\addcontentsline{toc}{chapter}{Bibliography}
\bibliographystyle{natbib}
\bibliography{references.bib}

% \appendix

% \chapter{Supervised Dimension Reduction}
% \subfile{chapters/appendix1}

\end{document}
