\documentclass[../thesis.tex]{subfiles}

\begin{document}

\begin{abstract}
In this work, we employ a variety of methods, some novel, from statistical and machine learning to genomics data with applications in prognostic and diagnostic medicine. We address applications of a variety of 'omics data types to these settings, and our aim in each application case is to inform practical decision making in the design and use of clinical tests. While we are principally concerned with clinical practicality, we also uncover results of independent scientific interest. 
We begin in Chapter~\ref{chap:intro} with an overview of the technologies and methodologies that are common to each of our investigations, including the basics of DNA and RNA biology (with a particular focus on cancer, the subject of Chapters~\ref{chap:tmb_estimation}~and~\ref{chap:causal_genomics}), Bayesian statistics, high-dimensional statistics, and causality.
In Chapter~\ref{chap:lamp_modelling} we present new analysis of the loop-mediated isothermal amplification (LAMP) assay on clinical and synthetic samples, motivated by the unmet need for statistcally principled methods for guided LAMP optimisation. In this context, optimisation refers to the selection of gene targets for profiling in blood-based diagnostic tests for sepsis, and to the tuning of assay conditions and selection of optimal primers to produce robust and high-resolution measurements of gene expression. 
We continue in Chapter~\ref{chap:tmb_estimation} with another study into the optimal, data-driven design of a biomedical test, albeit in a different clinical setting. Here, we are concerned with the prediction of tumour mutation burden (TMB), a key clinical biomarker determing how likely cancer patients are to respond to immunotherapy. Our task is to select a small number of gene targets to form a targeted DNA sequencing panel from which to estimate TMB. 
Finally, in Chapter~\ref{chap:causal_genomics}, we extend the previous chapter's work, and investigate the extent to which panel-based genomic markers can be tailored to identify heterogeneous causal effects in immunotherapy response.


% While somatic mutations have been understood for a long time as the instigators and drivers of tumourigenesis, it is now well known that the tumour environment and external factors play key roles in cancer development and spread. It is therefore of key importance to appreciate the nature of this balance. We formulate our interrogation of this broad question as a signal-to-noise problem, and attempt to ascertain to what extent the downstream properties of tumours can be predicted purely on the basis of the mutations they carry. This is in turn directly applicable to a growing area of clinical practice, liquid biopsy. Unlike solid biopsy, the nature of liquid biopsy means that we are only able to sequence tumour DNA, requiring any decisions based on liquid biopsy to be based purely on somatic mutations, potentially for some subset of the genome. We present a blueprint for the design, development and estimation of clinical biomarkers of response to immunotherapy based on generative models of the genome-wide landscape of molecular processes in tumour cells. From such models we can also make inferences about the underlying mechanisms leading to genomic instability, and how the hypermutated phenotype interacts with the body's natural defences against cancer. 
\end{abstract}

\end{document}