\documentclass[../thesis.tex]{subfiles}

\begin{document}

\section*{Overview}
\Gls{lamp} is a fast and cost-effective technique for detection of \glsxtrshort{dna}/\glsxtrshort{rna}. Its lack of reliance on complex laboratory equipment makes it ideal for use in point-of-care biomedical applications such as the diagnosis of time-critical disease. In this chapter we investigate the use of \gls{lamp} for profiling gene expression by measuring \gls{mrna} in blood samples. This is used in a test to establish whether patients with acute inflammation are bacterially or virally infected.  

While \gls{lamp} has many appealing properties, it remains unclear what factors affect its quantitative resolution. A lack of model-based frameworks to characterise \gls{lamp} data presents an unmet need in enabling the development and use of the assay. At present each of the assay's multiple primers are typically optimised experimentally, presenting a major bottleneck in assay design.  
We present hierarchical Bayesian models of LAMP amplification based on Gompertz functions, and use these models to infer the effect of \glsxtrshort{rna} variation and other factors on \gls{lamp} amplification curves derived from $\sim$100 blood samples of patients with suspected acute infection. 

Our analysis uncovers associations between LAMP assay resolution and characteristics such as primer sequence composition and thermodynamic properties. In addition to correlations between \glsxtrshort{rna} input abundance and time shift of the LAMP amplification curve, we also detect \glsxtrshort{rna}-dependent assocations with amplification rate.  We further investigate associations between primer/target properties and quantitative performance of the assay by generating a set of synthetic \glsxtrshort{rna} samples with systematically varied primer sequences and applying our framework. We find evidence that the associations observed are driven by across-target rather than within-target variation. This has important implications for study design, where in the past similar analyses have relied upon datasets including only one primer configuration per target. Our findings represent first steps towards data-driven development of quantitative assays, a key need in enabling \gls{lamp}'s use in the range of application spaces for which it shows promise.


\section{Introduction \label{sec:intro}}

\Glsfirst{lamp} is a fast, sensitive, and precise method for detecting \glsxtrshort{dna} or \glsxtrshort{rna}, the latter case referred to as reverse-transcription (RT)-LAMP. Since its introduction~\citep{notomi_loop-mediated_2000}, \gls{lamp} has been used for pathogen detection~\citep{mekata_real-time_2009, cao_development_2017, thiessen_development_2018}, sex identification~\citep{hirayama_embryo_2013, almasi_loop_2017, centeno-cuadros_validation_2018}, and cancer monitoring~\citep{li_loop-mediated_2016, horiuchi_novel_2020, kalofonou_novel_2020}. 
Unlike comparable assays such as \gls{pcr}, \gls{lamp} is isothermal and therefore does not require a thermocycler. This makes the assay particularly amenable to point-of-care and field applications~\citep{fu_applications_2011}. In particular, we study the use of \gls{lamp} in diagnostic testing for time-critical acute medicine, a setting in which lab-based \gls{pcr} is unviable.

We begin by introducing some terminology. We refer to a \emph{sample} as the material to be profiled for a given patient, in our case from a blood sample. For each sample, we aim to profile the expression of multiple \glsxtrshort{rna} sequences, referred to as \emph{targets}.  Each target will typically be chosen to represent a particular gene of interest, but could be any sequence. The \gls{lamp} assay comprises an amplification reaction initiated by the target sequence and whose progress can be measured for detection/quantitation. The reaction in question relies on several \emph{primers}. These are oligonucleotides (short sequences of \glsxtrshort{dna}) that are complementary to subsequences of the target and bind to them by \emph{hybridisation}. Once these primers have bound to the target or to subsequent products, they act as initation points for polymerisation, leading to the production of more nucelotide sequences which fold and separate to form an \emph{amplicon}. This amplicon then serves as the basis for further amplification steps. The components of the reaction are chemically tagged such that amplification can be measured via \emph{fluorescence}. By this we mean that the reaction emits light that can be measured and forms the basis for the data we aim to analyse.

As in \gls{pcr}, \gls{lamp} specificity relies on primer-target complementarity. In its original conception, two pairs of primers (referred to as F3/B3 and FIP/BIP) were proposed, and further optional pairs of primers were introduced for loop hybridisation~\citep{nagamine_accelerated_2002} and amplicon enlargement~\citep{gandelman_loop-mediated_2011}. In the experiments discussed here, six primers are used: the core pairs F3/B3 and FIP/BIP, and a loop hybridisation pair LF/LB.  A technical overview of the \gls{lamp} reaction is given in Figure~\ref{fig:lampoverview}. It is not necessary to understand every step of the reaction to follow the analysis presented in this chapter; readers need only follow that the \gls{lamp} assay requires the design of six primers. The need for coordinated design of multiple primers can create a major bottleneck in \gls{lamp} assay development.

\begin{figure}[!tpb] 
\centering
\includegraphics[width=5in]{../figures/chapter2/paper_lamp_fig.png}
\caption{Intermediate steps in the \glsxtrshort{qrtlamp} reaction. \textbf{A}: Following reverse transcription, any of the primers F3, FIP (comprising F2 and F1c sequences), and LB may anneal to the cDNA template. Polymerisation from the bound FIP primer leads to introduction of the F1c sequence into the product. \textbf{B}: Annealing of the F1c sequence to its complement creates a loop. The primers B3, BIP (comprising B2 and B1c sequences), and LF anneal to this single-loop template. \textbf{C}: Annealing of the B1c sequence to its complement creates another loop, resulting in the canonical `dumbbell' \gls{lamp} template, or amplicon. \textbf{D}: The stem region of the dumbbell template (the span between F1c and B1 sequence regions) contains the target sequence of interest while the loop regions (the regions spanning from F1c/B1 inclusive to F1/B1c exclusive) provide additional sites to initiate polymerisation. Crucially, any of the six primers may anneal and initiate polymerisation at any point at which a complementary region is available, leading to creation of other products (not shown for simplicity) distinct from the canonical product.\label{fig:lampoverview}}
\end{figure}
While \gls{lamp} has primarily been used in qualitative/detection applications, advances in measurement technology~\citep{zhang_brief_2014, becherer_loop-mediated_2020}, in particular automated fluorescence detection for producing \gls{lamp} amplification curves, have enabled quantitative use of the assay. Previous work has extended this approach to measurement of \gls{mrna} in a given sample, a technique referred to throughout as \gls{qrtlamp} \citep{remmel_diagnostic_2022}. Quantification for \gls{qrtlamp} requires analysis of the same data (one amplification curve per sample per target), but rather than detecting simply whether a reaction has initiated aims to extract quantitative data reflective of the input \glsxtrshort{rna} abundance.

Current understanding of the ideal properties of primers and targets to optimise \gls{qrtlamp}'s performance is incomplete and qualitative; primer design relies on a small set of guiding principles~\citep{panno_loop_2020}, including restrictions on the GC nucleotide content, melting temperature, and presence of GG repeats in primer sequences. In practice, primer design proceeds in trial-and-error fashion, as these guidelines can often be in conflict and are far easier to simultaneously satisfy for some \glsxtrshort{rna} targets than others. Furthermore, there has been little work to assess the impact on performance of sequence-based and thermodynamic properties of the \gls{lamp} primers and amplicon.

Statistical modelling frameworks provide one approach to link assay properties and characteristics of amplification curves. Extensive work exists for modelling both \gls{pcr} amplification curves~\citep{spiess_highly_2008, matz_no_2013, subramanian_empirical_2014,  nguyen_quantification_2020} and the effect of primer/target properties on \gls{pcr} assay performance~\citep{mallona_pcrefficiency_2011, wright_exploiting_2014, doring_modeling_2019}. Previous work on primer design for \gls{pcr} has identified thermodynamic properties of primer/target interactions~\citep{mann_thermodynamic_2009, li_primer_2010, doring_modeling_2019} and sequence length~\citep{huang_developing_2022} as key quantities associated with assay specificity. Thermodynamic properties (e.g. free energy of primer-target hybridisation) have also been associated with the speed of isothermal reactions~\citep{kimura_optimization_2011}.
% Direct mechanistic modelling of \gls{qpcr} has also been applied successfully \citep{boggy_mechanistic_2010}. 
In contrast, the study of \gls{lamp}-based quantitation is still under active development, and methods for \gls{qrtlamp} data analysis are lacking. 

Gompertz functions~\citep{gompertz_xxiv_1825} have proved a useful tool for empirical modelling of growth curves in varied biological settings~\citep{tjorve_use_2017}.  One formulation of the Gompertz curve is parameterised by an `amplitude' (the maximum value reached at asymptote), a `rate', and an `offset' (the horizontal shift of a curve along its time axis). Gompertz functions are similar to logistic functions, but allow for steeper increases from baseline and slower approaches to asymptote than logistic functions. In Section~\ref{sec:curvenormalisation} we define Gompertz functions formally (Equation~\ref{eq:gompertzcurve}) and give some examples alongside real amplification curves to demonstrate their properties (Figure~\ref{fig:gompertz_fit}). Bayesian and hierarchical Bayesian formulations of Gompertz curves have appeared in biomedical applications~\citep{wiper_bayesian_2010,  demirhan_bayesian_2015, sasaki_informative_2016,gotuzzo_bayesian_2019, vaghi_population_2020, berihuete_bayesian_2021} and have started to appear in LAMP reaction modelling for detection~\citep{carvalho_faster_2021}. However, no Bayesian or hierarchical methods have yet been applied to characterise \gls{qrtlamp} dynamics or to model the effects of different properties of \gls{lamp} targets/primers on quantitative assay performance.

Here we present a model-based analysis of the quantitative characteristics of the \gls{lamp} assay. We first develop a Bayesian modelling framework for inferring the impact of changes in \glsxtrshort{rna} input on amplification rate and offset, and we apply this framework to a dataset of \gls{qrtlamp} curves derived from 109 patients suspected of acute infection. Then, we compile a collection of primer/target features, and extend our hierarchical framework to detect associations between these features and resolution (or the ability to detect differences in \glsxtrshort{rna} input) of the \gls{qrtlamp} assay. In order to further investigate the effects of primer/target features on assay resolution, we utilise a novel dataset from synthetic \gls{ivt} \glsxtrshort{rna} templates assayed with systematically varied \gls{lamp} primer sequences. This dataset allows us to apply our framework in a context where we apply multiple primer sets to each target, and in so doing more directly interrogate the effects of changes to primer design. Our model-based analysis highlights several key considerations in the design and optimisation of the \gls{qrtlamp} assay, including the use of amplification rate for quantitative inference and the inclusion of thermodynamic properties into decisions around primer design. We provide a description of our implementation of this work via the R package \texttt{LAMPPrimerFeatures} and a \texttt{targets} analysis workflow in Appendix~\ref{appchap:lamp_code}.

% In order to understand the effects of target abundance on the characteristics of \gls{qrtlamp} curves, we model the parameters of a Gompertz curve corresponding to cycle offset and maximum rate as linear functions of \glsxtrshort{rna} count. We then quantify the impact of primer and amplicon design by including interaction terms for primer/target properties. We combine these features into hierarchical Bayesian model for simulation and inference. Multi-level effects allow us to model the dependence of an assay's \glsxtrshort{rna} resolution on primer/target-specific thermodynamic properties. Through our hierarchical modelling we take first steps towards establishing primer/target specific quantities of interest for assay design, suggesting the potential of our approach for refining primer design workflows, as well as the more general problem of target selection.
 

\enlargethispage{12pt}

% Equation~(\ref{eq:01}) 
% Figure~2\vphantom{\ref{fig:02}} 
% \citealp{Boffelli03}
% .....
% \begin{equation}
% \sum \text{\it x}+ \text{\it y} =\text{\it Z}\label{eq:01}\vspace*{-10pt}
% \end{equation}

\section{Methods}
\subsection{Datasets and technologies \label{sec:data}}
We utilised two datasets, one derived from clinical samples and one from synthetic \gls{ivt} \glsxtrshort{rna} templates. We summarise the data availability for each of the two experimental setups in Table~\ref{tab:datasets}. 

Our clinical dataset comprised blood samples collected from 109 patients, each adjudicated to be either bacterially infected, virally infected, or non-infected (with these three labels occurring in approximately equal proportions). For each sample, we analysed gene expression measured from 28 target \glsxtrshort{rna} transcripts: 25 genes previously identified as targets of interest for bacterial/viral/non-infected classification~\citep{he_optimization_2021} as well as 3 \emph{housekeeping} genes (KPNA6, RREB1, and YWHAB). By a housekeeping gene, we mean one whose expression is not expected to vary substantially between samples, and so can act as reference for normalisation. We tracked fluorescence for 20 minutes or 60 `cycles' (1 cycle = 20 seconds)\footnote{Note that these are not strictly cycles since the \gls{lamp} reaction is isothermal, but the cycle nomenclature is still common in the \gls{lamp} literature.}. For each sample/target, two technical replicates were performing using separate subportions (\emph{aliquots}) of the given sample. Another aliquot was used for \gls{mrna} quantification using the \gls{ns} nCounter system. The \gls{ns} nCounter system  directly counts \glsxtrshort{rna} transcript number in a single step by measurement of fluorescence via hybridisation of capture and reporter probes. As such, \gls{ns} measurements are a singular value (unlike other comparison technologies such as \gls{qpcr}) and very sensitive even at low expression levels \citep{geiss_direct_2008}. While \gls{ns} is unsuitable for point-of-care usage, it is an ideal technology to provide `gold standard' benchmark measurements. Higher \gls{ns} values correspond to higher target expression. 

Our \gls{ivt} \glsxtrshort{rna} dataset (preparation described in~\citealp{remmel_diagnostic_2022}) comprised measurements for three targets: {IFI27}, {JUP}, and {OASL}. This dataset was generated by assaying different concentrations of synthetic \glsxtrshort{rna} templates corresponding to each of the the targets. To test the effects of varying different primer properties on \gls{qrtlamp} performance, FIP and BIP primer sequences were varied for each target. These sequence alterations consisted of variations in the lengths of annealing sequence for either the F1c (B1c) or F2 (B2) subsequences of the FIP (BIP) primer (Figure~\ref{fig:lampoverview}). F3, B3, LF, and LB primer sequences were not varied. For each target/primer set combination, \gls{qrtlamp} measurements at three \glsxtrshort{rna} concentrations ($10^1$, $10^3$, and $10^5$ copies/$\mu$L; 20$\mu$L per reaction) were collected, with at least two replicates per concentration. We investigated effects of changes in FIP/BIP primer sequence in 35, 48 and 63 primer combinations for {IFI27}, {JUP}, and {OASL} respectively. For the \gls{ivt} samples, fluorescence was tracked for 30 minutes, or 90 cycles. 
\begin{table}[!t] 
\centering
{\begin{tabular}{ccc}
\hline \textbf{Summary} & \textbf{Clinical Dataset} &
\textbf{IVT Dataset}  \\\hline
Number of samples & 109 & 3  \\
Number of targets  & \multirow{2}{*}{25} & \multirow{2}{*}{3} \\
(non-housekeeper) & & \\
Number of targets & \multirow{2}{*}{3} & \multirow{2}{*}{0} \\
(housekeeper) & & \\
Replicates per sample/target & \multirow{2}{*}{2} & \multirow{2}{*}{$\geq 2$} \\
(qRT-LAMP) & & \\
Primer sets per sample/target & 1 & 35-63 \\
Replicates per sample/target & \multirow{2}{*}{1} & \multirow{2}{*}{0} \\
(NS) & & \\
Cycles per run & 60 & 90 \\
\hline
\end{tabular}}
\caption{Specification of clinical and IVT datasets.\label{tab:datasets}} 
\end{table}

We now describe technical aspects of sample preparation and collection for each of the datasets used. Blood samples from our clinical dataset were collected using the PAXgene \glsxtrshort{rna} system, as described in \citet{ram-mohan_using_2022}. These patients samples were drawn from patients involved in two clinical trials: ``HostDx Sepsis in the Diagnosis and Prognosis of Emergency Department Patients With Suspected Infections: a Multicenter Pilot Study'' (\url{clinicaltrial.gov/NCT03744741}), and ``Efficiency in Management of Organ Dysfunction Associated with Infection by the Novel SARS-CoV-2 Virus Through a Personalised Immunotherapy Approach: The ESCAPE Clinical Trial: The `ESCAPE' Trial (a.k.a ESCAPE)'' (EudraCT/2020-001039-29). At enrollment, patient blood was collected and immediately stored with PAXgene \glsxtrshort{rna} stabilisation. Two independent aliquots of each sample underwent \glsxtrshort{rna} extraction for profiling by \gls{qrtlamp} and were dispensed along with primers and reagents and run on a 96-well plate with a QuantStudio\texttrademark~5 Real-Time PCR System for Human Identification (Applied Biosystems; Waltham, MA USA).  We normalised \gls{ns} measurements by first rescaling counts such that housekeeping genes had a geometric mean of 1000 and then applying a \textsf{log}$_{2}$ transformation to the rescaled counts. To measure \gls{mrna} expression with \gls{qrtlamp}, a set of six \gls{lamp} primers (FIP/BIP,~F3/B3,~and  LF/LB;~Figure~\ref{fig:lampoverview}) for each target were designed using PrimerExplorer v5~\citep{eiken_primerexplorer_2019} or via a third-party service provider (Primer Digital, Ltd; Helsinki, Finland). \Gls{qrtlamp} reactions occurred at $65^{\circ}$C, in the presence of 90mM KCl and 8mM MgSO$_4$. Originally 29 non-housekeeper targets were measured, but we excluded 4 for technical reasons: {FURIN} and {S100A12} due to the use of different primer concentrations; {CTSB} and {DEFA4} as LF/LB primers did not anneal to their target sites. Reaction conditions for \gls{qrtlamp} with \gls{ivt} \glsxtrshort{rna} templates were identical to those used for the clinical samples. Replicate samples were dispensed along with primers and reagents and run on a 384-well plate with the QuantStudio\texttrademark~5 instrument. 

% this stuff below should probably go in the introduction instead
% By contrast, the qRT-LAMP assay works by reverse transcription of target
% \glsxtrshort{rna}, followed by attachment of a sequence of primers which form a double looped `dumbbell’ structure that is particularly amenable to hybridisation and subsequent amplification as described in Section~\ref{sec:intro}. 
% A fluorescent tag is incorporated into amplicons, and changes in fluorescence over time are measured. 
% \gls{lamp}'s high degree of complexity and parallelism means that variation can be introduced at each of the stages described, and can occur at the level of targets, biological samples, and technical replicates. 

\subsection{Amplification curve pre-processing and normalisation \label{sec:curvenormalisation}}
To prepare \gls{qrtlamp} curves for analysis, we performed a series of pre-processing steps. We first applied baseline correction to set the initial fluorescence of each reaction approximately to zero. 
% Signal baseline was estimated by identifying a time interval with near-linear signal (2nd differential close to 0), then fitting a line with a Theil-Sen estimator. 
% After correcting the signal baseline, we removed time points with negative fluorescence from each reaction curve.
We then used a nonlinear least-squares approach to fit Gompertz functions to each amplification curve. The Gompertz function is parameterised by three quantities ($a$, $b$, and $c$ in Equation~\ref{eq:gompertzcurve}) associated with the amplitude, rate, and cycle offset, respectively, of the amplification curve. One can think of the parameter $c$ as a time-to-threshold measurement (as used for \gls{qpcr}) in which the threshold is specific to each reaction curve. Higher cycle offset (a right-shifted curve) would indicate lower amounts of \glsxtrshort{rna}, and vice versa. The Gompertz function is defined as follows:
\begin{equation} \label{eq:gompertzcurve}
    y = a \exp\Big(-\exp\big( -b (t - c)\big)\Big).
\end{equation}
Here, $y$ corresponds to baseline-corrected fluorescence and $t$ is time in cycles. Our procedure estimates parameters $(\hat{a}^{\text{gomp}}, \hat{b}^{\text{gomp}}, \hat{c}^{\text{gomp}})$ for a given amplification curve. Anecdotally and experimentally, \gls{lamp} amplification curves often become very noisy around asymptote, and for reasons of technical calibration the absolute level of asymptotic fluorescence ($a$) is relatively unimportant (as with qRT-\gls{pcr}) to quantitation compared to cycle offset ($c$). The impact of input \glsxtrshort{rna} abundance on amplification rate ($b$) is, by contrast, fairly unexplored.
 Motivated by this and after noting some misfit to the asymptote in early versions of our curve-fitting procedure, we developed and applied an iterative procedure, described in Algorithm~\ref{alg:normalise}, to remove observations exceeding the fitted asymptote at each iteration. We then retained the fitted asymptote ($\hat{a}^{\text{gomp}}$) at the last iteration (when no points of the curve exceeded the asymptote) to rescale the curve. A selection of example outputs from this procedure appear in Figure~\ref{fig:gompertz_fit}.

\begin{algorithm}[ht]
\SetAlgoLined

\KwData{The set $\{(t,D_t): t = 1, \dots, T\}$, where $T$ is the number of cycles observed and $D_t$ is the fluorescence measured by the \glsxtrshort{qrtlamp} assay at cycle $t$.} 

\KwOut{A triple $(\hat{a}^{\text{gomp}}, \hat{b}^{\text{gomp}}, \hat{c}^{\text{gomp}})$ of Gompertz parameter estimates, and a pruned set $\{(t,D_t): t = 1, \dots, n^*\}$.}

Set $n = T$; \\
Set $\delta = 0$; \\
 \While{$(\delta \geq 0)$}{
    Set $(\hat{a}, \hat{b}, \hat{c}) = \argmin\limits_{{(a,b,c) \in \mathbb{R}^3}} \big\{ \sum_{t=1}^{n} \Big(D_t - a \exp\Big(-\exp\big( -b (t - c)\big)\Big)\Big)^2 \big\}$;\\ 
   
    Set $\delta = \max \{D_t: \ t \in \{1, \dots, n\} \} > \hat{a}$;\\
    Set $n = \min \{t: D_{t+1} > \hat{a}\};$
}
Set $(\hat{a}^{\text{gomp}}, \hat{b}^{\text{gomp}}, \hat{c}^{\text{gomp}}) = (\hat{a}, \hat{b}, \hat{c})$; \\
Set $n^* = n$. \\
~\\
 \caption{Iterative least-squares Gompertz curve-fitting procedure \label{alg:normalise}}
\end{algorithm} 

\begin{figure}[!tpb] 
\centering
\includegraphics[width=6in]{../figures/chapter2/paper_gomp_quality_fig.png}
\caption{The normalisation procedure described in~\ref{sec:curvenormalisation}, with examples from our clinical dataset. \textbf{A}: Baseline-corrected \gls{lamp} curves with points corresponding to individual fluorescence measurements (arbitrary units) over 20s cycles. \textbf{B}: Rescaled and pruned \gls{lamp} amplification curves. Points show scaled measurements while solid blue lines correspond to Gompertz curve fits from our iterative procedure. \textbf{C}: Distribution of goodness-of-fit (measured by mean squared error) for our fitted Gompertz curves, with 50\%, 90\% and 99\% quantiles indicated by dashed vertical lines. Representative fits from these quantiles are shown in {A} and {B}. \textbf{D}: Reproduction of the third figure of panel {B}, with a logistic curve fit for comparison (red).  \label{fig:gompertz_fit}}
\end{figure}

\subsection{Single-target model \label{sec:singletargetmodel}}
To investigate the effects of \glsxtrshort{rna} abundance on \gls{qrtlamp} amplification curves for individual targets, we model the Gompertz curve parameter estimates produced by our nonlinear fitting procedure. For samples $i = 1, \dots, N$, replicates $j = 1, \dots, J$, with \gls{ns}-measured gene expression $x_i$ (a proxy for true \glsxtrshort{rna} input in clinical samples), we model estimated parameters for rate ($\hat{b}_{ij}^{\text{gomp}}$) and cycle offset ($\hat{c}_{ij}^{\text{gomp}}$) of an individual target as realisations of independent random variables $\hat{B}_{ij}$ and $\hat{C}_{ij}$ distributed according to:
\begin{align} 
    \hat{B}_{ij} & \sim \textsf{Student-t}(\mu_B + \lambda_B x_i + \rho_B \hat{h}^{\text{gomp}}_i, \ \nu_B, \ \sigma_B ), \label{eq:bsm}\\
    \hat{C}_{ij} & \sim \textsf{Student-t}(\mu_C + \lambda_C x_i + \rho_C \hat{h}^{\text{gomp}}_i, \ \nu_C, \ \sigma_C). \label{eq:csm}
\end{align}
\Gls{pgm} plate diagrams for these models are given in Figure~\ref{fig:stm_pgms}. We use a $\textsf{Student-t}(\mu,\nu,\sigma)$ likelihood to provide robustness to outliers, with $\mu$, $\nu$, and $\sigma$ corresponding to the location, shape, and scale of the distribution respectively. We refer to $\lambda_B$ and $\lambda_C$ as resolution parameters since a larger absolute value of $\lambda_B$ or $\lambda_C$ would reflect greater sensitivity, on average, to changes in \glsxtrshort{rna} input for a given target.
Note that these properties are with respect to absolute changes in \glsxtrshort{rna} input, not changes in \glsxtrshort{rna} input relative to the overall variation of a given target. We are therefore specifically attempting to measure \gls{lamp}'s technical performance measuring a biochemical target, not the total discriminatory power of the range of a target's expression. When, for example, selecting targets for a diagnostic test, other considerations such as clinical significance and variation in expression will have to be weighed against technical resolution of the assay.

\begin{figure}
    \centering
    \includegraphics[width=.45\textwidth]{figures/chapter2/LAMP_STM_B_PGM.png} \includegraphics[width=.45\textwidth]{figures/chapter2/LAMP_STM_C_PGM.png}
    \caption{PGM plate diagrams for single-target models of $\hat{B}$ (left) and $\hat{C}$ (right).}
    \label{fig:stm_pgms}
\end{figure}
%$\lambda_B$ represents the rate of fluorescence increase, similar to the efficiency for \gls{pcr} reactions~\citep{svec_how_2015}. Similarly, $\lambda_C$ is comparable to the $C_{t}$ (time-to-threshold) measure of gene expression found in \gls{pcr}. 
The input $\hat{h}^{\text{gomp}}_{i}$ is the geometric mean of the values $\hat{c}_{ijk}^{\text{gomp}}$ across all replicates $j$ and all housekeeping genes  $k \in$ \{KPNA6, RREB1, YWHAB\}, included to normalise for sample-to-sample variation in \glsxtrshort{rna} yield. The parameters $\sigma_B$ and $\sigma_C$ capture variation in rate and cycle offset that is not attributable to variation in \glsxtrshort{rna} input. Bayesian inference is performed with the R package \texttt{brms}~\citep{burkner_brms_2017}. We specify priors as follows. For the rate model (eq.~\ref{eq:bsm}): $\mu_B \sim \mathcal{N}(0.5, 0.5^2)$, $\lambda_B \sim \mathcal{N}(0, 0.1^2)$, $\rho_B \sim \mathcal{N}(0, 0.1^2)$, $\nu_B \sim \Gamma(2,0,1)$, $\sigma_B \sim \textsf{Student-t}(3, 0, 2.5^2)$; for the cycle offset model (eq.~\ref{eq:csm}): $\mu_C \sim \mathcal{N}(30, 5^2), \lambda_C \sim \mathcal{N}(0, 1^2), \rho_C \sim \mathcal{N}(0, 1^2)$, and identical priors to the rate model for $\nu_C,\sigma_C$. We approximate model posteriors with 4000 samples (1000 warm-up) from 3 \gls{hmc} chains, assessed for convergence with all parameter posteriors achieving $\hat{R} \leq 1.01$ and bulk and tail effective sample sizes above 1000.

\subsection{Multi-target model \label{sec:hierarchicalmodel}}
To evaluate associations between primer/target properties and assay resolution, and to model \gls{qrtlamp} amplification curves across all targets, we propose hierarchical models of the estimated rate and cycle offset. We derive estimates of the rate $\hat{b}^{\text{gomp}}_{ijkl}$ and offset $\hat{c}^{\text{gomp}}_{ijkl}$ for sample $i$, replicate $j$, target $k$, and primer set $l$ (only varied in the case of \gls{ivt} \glsxtrshort{rna} templates) from our non-linear least squares fitting procedure. We then model each of the $\hat{b}^{\text{gomp}}_{ijkl}$ and $\hat{c}^{\text{gomp}}_{ijkl}$ as being independent realisations of random variables $\hat{B}_{ijkl}$, $\hat{C}_{ijkl}$, where: 
\begin{align} 
    \hat{B}_{ijkl}  & \sim  \textsf{Student-t}(\mu_B(\bm{z}_{kl}; \bm{\beta}_B) + \lambda_B(\bm{z}_{kl}; \bm{\gamma}_B) x_i + \rho_B \hat{h}^{\text{gomp}}_{i}, \ \nu_B, \ \sigma_B), \label{eq:bhm} \\
    \hat{C}_{ijkl}  & \sim  \textsf{Student-t}(\mu_C(\bm{z}_{kl}; \bm{\beta}_C) + \lambda_C(\bm{z}_{kl}; \bm{\gamma}_C) x_i + \rho_C \hat{h}^{\text{gomp}}_{i}, \ \nu_C, \ \sigma_C). \label{eq:chm}
\end{align}
Here, for each model the location of the likelihood comprises \glsxtrshort{rna}-independent baseline offset $\mu_B$ (/$\mu_C$) and \glsxtrshort{rna}-dependent resolution $\lambda_B$ (/$\lambda_C$) functions, parameterised by $\bm{\beta}_B$ (/$\bm{\beta}_C$) and $\bm{\gamma}_B$ (/$\bm{\gamma}_C$), respectively. These functions (which we will refer to as $\mu$ and $\lambda$ when speaking interchangeably about either model) take as an input a vector of sequence-based and thermodynamic features, $\bm{z}_{kl}$, associated with the given primer set $l$ and target $k$. In our clinical dataset, $\bm{z}$ has no dependence on $l$ as each target corresponds to a single primer set. Also, for our clinical dataset, we let baseline offset $\mu$ vary across targets $k$ by specifying it as a hierarchical random effects term. We also remove dependence of $\mu$ on $\bm{z}_{k}$:
\begin{eqnarray} \label{eq:muclin}
    \mu(\bm{z}_k; \bm{\beta}) &=& \beta_{0,k}, \\
    \beta_{0,k} &\sim& \mathcal{N}(\mu_{\beta_{0}},\sigma^{2}_{\beta_{0}}).
\end{eqnarray}
This random term models target-specific effects unrelated to assay characteristics. Our models of baseline offset in clinical samples did not have slopes ($\beta_{z}$) varying by target as we had only one primer set per target. 

For our \gls{ivt} \glsxtrshort{rna} dataset, we note that: \emph{a)} $x_{i}$ represents the log \glsxtrshort{rna} copy number at concentration $i$, and \emph{b)} we drop the housekeeping term ($\rho\hat{h}^{\text{gomp}}_{i}$) as input \glsxtrshort{rna} concentrations are known. For both our \gls{ivt} \glsxtrshort{rna} and clinical data, we use $\lambda$ to model effects of assay features on (RNA-dependent) assay resolution (e.g. $\lambda(\bm{z}_{kl}; \gamma) = \gamma_{0} + \gamma_{L_{\text{stem}}} L_{\text{stem},k,l}$).

For analysis of our \gls{ivt} \glsxtrshort{rna} dataset, we specify two types of models, which we refer to as \emph{across-target} and \emph{within-target} models respectively. 
% The first model type, \emph{across-target}, is similar to the multi-target model used for clinical samples (Equation~\ref{eq:chm}) except that 
For both model types, the \glsxtrshort{rna}-independent offset function, $\mu$, includes random intercepts and random slopes to model dependence on assay properties. For example, when using stem length as an input: $\mu(\bm{z}_{kl}; \bm{\beta}) := \beta_{0,k} + \beta_{L_{\text{stem},k}} L_{\text{stem},k,l}$. This is to allow maximum flexibility in the \glsxtrshort{rna}-independent offset to focus on changes in resolution (via $\lambda$) due to primer set features.
The form of the resolution function $\lambda$ for our \emph{across-target} model is identical to that applied to clinical data, while for the \emph{within-target} model, we include random target-specific intercepts in the resolution function $\lambda$. Again, using stem length as an example, the \emph{within-target} model $\lambda$ function would be:
\begin{eqnarray}
\lambda(\bm{z}_{kl}; \bm{\gamma}) &=& \gamma_{0,k} + \gamma_{L_{\text{stem}}} L_{\text{stem},k,l}, \\
\gamma_{0,k} &\sim&  \mathcal{N}(\mu_{\gamma_{0}}, \sigma^{2}_{\gamma_{0}}).    
\end{eqnarray}
%For both `global' parameters such as $\gamma_0$ and $\gamma_{L_s^{\text{amp}}}$ and hierarchical parameters such as $\mu_\gamma$ we used \text{Student-t}s, such that for example $\gamma_0 \sim \textsf{Student-t}(\mu_{\gamma_0}, \nu_{\gamma_0}, \sigma_{\gamma_0})$.

\begin{figure}
    \centering
    \includegraphics[width=.45\textwidth]{figures/chapter2/LAMP_ATM_B_PGM.png} \includegraphics[width=.45\textwidth]{figures/chapter2/LAMP_ATM_C_PGM.png} \\
    \vspace{10pt}
    \includegraphics[width=.45\textwidth]{figures/chapter2/LAMP_WTM_B_PGM.png} \includegraphics[width=.45\textwidth]{figures/chapter2/LAMP_WTM_C_PGM.png}
    \caption{PGM plate diagrams for $\hat{B}$ (left) and $\hat{C}$ (right). Specifications are given for multi-target models with across-target resolution functions (top row) and within-target resolution functions (bottom row).}
    \label{fig:mtm_pgms}
\end{figure}

We give \gls{pgm} plate diagrams for both across- and within-target models for $\hat{B}$ and $\hat{C}$ in Figure~\ref{fig:mtm_pgms}. Note that the only difference is whether the parameter $\gamma_0$ is `inside' the plate associated with target $k$, i.e. is allowed to vary between targets. When the $\lambda$ function parameter $\bm{\gamma}$ is a `global' parameter (i.e. not specified as a hierarchical term), we use independent $\mathcal{N}(0,2^2)$ priors for its component parameters. When $\bm{\gamma}$ includes hierarchical terms for the intercept, we specify priors $\mu_{\gamma_{0}} \sim \mathcal{N}(0, 2^2)$, $\sigma_{\gamma_{0}} \sim \textsf{Student-t}(3, 0, 2.5^2)$. Priors for the baseline offset parameters $\bm{\beta}$ are set similarly, aside from the intercept term $\beta_0$: $\mu_{\beta_0} \sim \mathcal{N}(1, 1^2)$ for our rate model and $\mu_{\beta_0} \sim \mathcal{N}(23, 5^2)$ for our cycle offset model. In each case the parameters $\nu$ and $\sigma$ are given priors $\nu \sim \Gamma (2, 0.1), \sigma \sim \textsf{Student-t}(3, 0, 2.5^2)$.
For multi-target model parameter inference, we use 3 \gls{hmc} chains of 4000 samples (1000 warm-up), again assessing convergence by confirming that all chains have $\hat{R} \leq 1.01$ and effective bulk and tail sample sizes above 1000.

\subsection{Properties of primers and targets \label{sec:freeenergy}}
Multiple properties are known to impact efficacy of \gls{lamp} primers, including melting temperature, GC content and GG repeat abundance~\citep{eiken_primerexplorer_2019}. In practice, many of these properties are tightly controlled in primer design workflows, resulting in little variability with respect to these quantities. We therefore focused on three classes of \gls{lamp} properties: \emph{a)} sequence features of the \gls{lamp} target or amplicon; \emph{b)} thermodynamic properties of primers and their targets; and \emph{c)} measures of primer sequence complexity. The full list of properties appears in Table~\ref{tab:properties}. 
\begin{table}[!t] 
\centering
{\begin{tabular}{cccc}
\hline \textbf{Property type} &\textbf{Property} & \textbf{Symbol} &
\textbf{Unit}  \\\hline
\multirow{3}{*}{Amplicon-specific} & Stem length & $L_s^{\text{amp}}$ & bp  \\
&Loop length & $L_l^{\text{amp}}$ & bp  \\
&GC Content & $GC^{\text{amp}}$ & -  \\
\hline
\multirow{6}{*}{Specific to primer $p$} &4-complexity & $C^p$ & -  \\ 
&Free energy of annealing & $\Delta G_a^p$ & kJmol$^{-1}$ \\
&Free energy of self- & \multirow{3}{*}{$\Delta G^p_{\text{sd}}$} & \multirow{2}{*}{kJmol$^{-1}$} \\
&dimerisation   & &  \\
&Free energy of secondary  & \multirow{2}{*}{$\Delta G^p_{\text{ss}}$} & \multirow{2}{*}{kJmol$^{-1}$} \\
&structure formation  & &  \\ 
\hline
\end{tabular}}
\caption{Properties of targets/primers used in hierarchical modelling of LAMP performance. Here $p \in \{F3, B3, FIP, BIP, LF, LB \}$.\label{tab:properties}} 
\end{table}
Stem length ($L_{\text{stem}}$) and loop length ($L_{\text{loop}}$) measure the total length in nucleotide bases of the amplicon stem and loop regions, respectively (Figure~\ref{fig:lampoverview}D). GC content ($GC$) is the proportion of bases of the entire amplification region (F3 to B3, inclusive) that are either G or C.

Each free energy property measures the strength of some binding process via Gibbs free energy. Free energy of annealing ($\Delta G_a^p$) measures the binding affinity of a primer with its complementary target region, while free energy of self-dimerisation ($\Delta G^p_{\text{sd}}$) quantifies a primer's propensity to anneal to another copy of the same primer. Finally, $\Delta G^p_{\text{ss}}$ measures a single primer's tendency to form secondary structure within itself. We follow the approach of~\citet{doring_modeling_2019}, using UNAFold~\citep{markham_unafold_2008} for free energy calculations and applying ion concentration adjustments as described in~\citet{peyret_prediction_2000}. For FIP (BIP) primers, we calculated separate free energies of annealing for F1c (B1c) and F2 (B2) primer subsequences.

To measure the extent to which a primer sequence contains unexpected $k$-mer sequences (i.e. sequences of length $k$) compared to a reference sequence, we also computed sequence complexity ($C^p$)~\citep{nielsen_design_2003, xia_evaluating_2010}. This measure is based on the information-theoretic notion of entropy~\citep{shannon_mathematical_1948}. We measure complexity with respect to 4-mers and use the entire human exome as contained in the \emph{Ensembl} database~\citep{yates_ensembl_2020} as our set of reference sequences. We use a simplified formulation of 4-complexity:
\[C^p_4 := 1 -  \frac{\sum_{\ell = 1}^{L-3} n^p_4(\ell) \log (n^p_4(\ell))}{Z}.\]
Here, $n^p_4(\ell)$ refers to the count of 4-mers found in the reference sequence set that match the 4-mer in primer $p$'s sequence starting at position $\ell$. The normalising constant $Z$ is set to ensure that complexity varies between 0 and 1. A low complexity score indicates that the majority of a primer's constituent $k$-mers occur commonly in the reference set.


\subsection{Bayesian stacking for model comparison and validation \label{sec:stacking}}
To compare multi-target models and identify which \gls{lamp} properties are most robustly associated with resolution, we use Bayesian model stacking based on Pareto-smoothed importance sampling approximations of each model's performance in leave-one-out cross-validation (LOO-PSIS; implemented by the R package \texttt{loo}; \citealp{vehtari_loo_2020}).

In traditional model stacking, predictions from multiple models are ensembled via a linear combination in order to minimise the average cross-validation prediction error~\citep{wolpert_stacked_1992}. The same process is applied in Bayesian model stacking, but rather than using the squared error of models' predictions, one calculates log predictive densities for left-out observed values~(i.e. those allocated to a validataion set; \citealp{yao_using_2018}). The outputs of stacking are model weights associated with each individual model in the ensemble. These weights will sum to one, and a larger weight indicates a more substantial contribution for the corresponding model (relative to other models in the ensemble) to the optimal model combination. Thus, we take large stacking weights to be indicative of stronger associations of the corresponding model's features with \gls{qrtlamp} assay resolution. We use Pareto-smoothed importance sampling to approximate~\citep{vehtari_practical_2017, vehtari_pareto_2021} the log posterior predictive density of each left out sample. Pareto $k$ diagnostic values $>0.5$ would indicate a poor approximation of the LOO posterior predictive density for some observations. We did not observe any such problematic Pareto $k$-estimates in our analyses.

\section{Results \label{sec:results}}

\subsection{RNA dependence for qRT-LAMP cycle offset and amplification rate varies widely between targets\label{sec:sensitivityprecision}}
Our first aim was to infer the resolution ($\lambda_B, \lambda_C$) and precision ($\sigma_{B}$, $\sigma_{C}$) parameters associated with \glsxtrshort{rna} input's effect on \gls{qrtlamp} curve characteristics in clinical samples. We used our single-target models (Equations~\ref{eq:bsm}~and~\ref{eq:csm}, Section~\ref{sec:singletargetmodel}) to decompose variation in cycle offset ($\hat{c}_{ij}$; median $22.2$ across all targets, \gls{iqr} $5.03$ across all targets) and rate ($\hat{b}_{ij}$; median $0.334$ across all targets, \gls{iqr} $0.076$ across all targets) into separate \glsxtrshort{rna}-dependent and \glsxtrshort{rna}-independent components. We fitted these models for each of our 25 targets of interest~\citep{he_optimization_2021}. Summaries of the posterior distributions for rate and cycle offset parameters appear in Tables~\ref{tab:single_target_offset_models_summary}~and~\ref{tab:single_target_rate_models_summary}. More in-depth summaries are given in Appendix~\ref{appchap:lamp_models}.

We see a wide range of inferred posterior means for both $\lambda_C$ and $\sigma_C$. For the parameter $\lambda_C$, which denotes resolution with respect to cycle offset, larger negative values correspond to improved resolution, and suggest that for a given change in RNA input a larger average change in amplification curve cycle offset will be observed. Offset precision $\sigma_C$ is always positive, with lower values corresponding to a more precise assay (i.e. less variation in amplification curve cycle offset between samples of the same RNA input). For example, targets {CTSL1} and {ZDHHC19} both show good resolution (posterior means of $-1.86$ and $-2.20$, 95\% credible intervals of (-2.09, -1.64) and (-2.59, -1.81), respectively), but different levels of precision (posterior means of $1.57$ and $6.15$, 95\% credible intervals of (1.42, 1.74) and (5.57, 6.80), respectively). Inferences of poor resolution and precision for some targets (e.g. {KIAA1370}) indicate the potential need for further optimisation of assay properties and/or conditions. We also note that, as expected and as reflected by the posteriors for $\rho_C$, the geometric mean of the housekeeping cycle offsets is generally correlated with the cycle offset of the target (i.e., lower sample yield should correspond to higher cycle offsets for all targets). 

The amplification rate parameter $\lambda_B$ also varied between targets, with a larger positive value indicating greater average sensitivity to changes in \glsxtrshort{rna} input. While some reactions, including for example {CD163} and {RAPGEF}, showed almost no detectable variation in rate due to \glsxtrshort{rna} input (posterior mean for $\lambda_B$ of 0.00, 95\% credible intervals of (0.00, 0.01) in both cases), others such as {PER1} and {ZDHHC19} did show a dependence on \gls{ns}-derived measurements of \glsxtrshort{rna} (posterior means for $\lambda_B$ of 0.04 in both cases, 95\% credible intervals of (0.03, 0.05) and (0.04, 0.04) respectively). Posterior summaries for the precision in amplification rate ($\sigma_{B}$) were roughly similar across targets. We also found that housekeeping levels showed a negative association with the rate of amplification (for example, 95\% posterior credible intervals for $\rho_B$ of (-0.03, -0.02) in ARG1 and (-0.02, -0.01) in BATF).



\begin{table}[ht]
\centering
\begin{tabular}{cccc}
  \hline
{\textbf{Target}} & $\bm{\lambda_C}$ & $\bm{\sigma_C}$ & $\bm{\rho_C}$ \\ 
  \hline
ARG1 & -0.71 (-0.78, -0.65) & 1.14 (1.02, 1.26) & 1.14 (1.02, 1.26) \\ 
  BATF & -1.02 (-1.16, -0.88) & 0.73 (0.66, 0.81) & 0.78 (0.70, 0.86) \\ 
  C3AR1 & -0.90 (-1.00, -0.79) & 0.88 (0.79, 0.97) & 0.54 (0.43, 0.65) \\ 
  C9orf95 & -0.64 (-0.85, -0.44) & 0.86 (0.78, 0.96) & 0.51 (0.41, 0.60) \\ 
  CD163 & -0.61 (-0.68, -0.53) & 0.51 (0.46, 0.56) & 0.36 (0.30, 0.43) \\ 
  CEACAM1 & -0.64 (-0.73, -0.56) & 0.97 (0.88, 1.08) & 0.42 (0.31, 0.54) \\ 
  CTSL1 & -1.86 (-2.09, -1.64) & 1.57 (1.42, 1.74) & 1.03 (0.86, 1.21) \\ 
  GADD45A & -0.68 (-0.80, -0.56) & 1.03 (0.94, 1.15) & 0.29 (0.17, 0.40) \\ 
  GNA15 & -0.77 (-0.93, -0.61) & 0.67 (0.60, 0.73) & 0.95 (0.85, 1.04) \\ 
  HK3 & -0.70 (-0.76, -0.63) & 0.44 (0.40, 0.49) & 0.71 (0.66, 0.76) \\ 
  HLA-DMB & -0.95 (-1.41, -0.51) & 3.25 (2.94, 3.61) & 0.67 (0.35, 0.98) \\ 
  IFI27 & -0.91 (-1.01, -0.80) & 2.10 (1.89, 2.34) & 0.69 (0.48, 0.89) \\ 
  ISG15 & -0.97 (-1.06, -0.87) & 2.10 (1.89, 2.34) & 0.84 (0.63, 1.06) \\ 
  JUP & -1.57 (-1.78, -1.36) & 2.43 (2.20, 2.69) & 1.51 (1.25, 1.77) \\ 
  KCNJ2 & -0.70 (-0.81, -0.60) & 0.75 (0.68, 0.83) & 0.18 (0.09, 0.27) \\ 
  KIAA1370 & -0.34 (-0.57, -0.10) & 0.99 (0.89, 1.10) & 0.42 (0.31, 0.53) \\ 
  LY86 & -0.71 (-0.85, -0.58) & 0.66 (0.60, 0.74) & 0.78 (0.70, 0.86) \\ 
  OASL & -0.83 (-0.91, -0.76) & 0.89 (0.80, 0.99) & 0.83 (0.74, 0.93) \\ 
  OLFM4 & -1.01 (-1.12, -0.90) & 1.58 (1.43, 1.77) & 0.52 (0.35, 0.68) \\ 
  PDE4B & -0.95 (-1.21, -0.69) & 1.35 (1.23, 1.49) & 0.90 (0.74, 1.05) \\ 
  PER1 & -1.41 (-1.74, -1.10) & 2.61 (2.36, 2.88) & 0.39 (0.20, 0.58) \\ 
  PSMB9 & -0.94 (-1.06, -0.83) & 0.66 (0.60, 0.73) & 0.88 (0.79, 0.96) \\ 
  RAPGEF1 & -0.60 (-0.95, -0.26) & 1.79 (1.62, 1.98) & 0.68 (0.51, 0.86) \\ 
  TGFBI & -0.71 (-0.79, -0.63) & 0.58 (0.53, 0.65) & 0.79 (0.72, 0.87) \\ 
  ZDHHC19 & -2.20 (-2.59, -1.81) & 6.15 (5.57, 6.80) & 0.89 (0.53, 1.26) \\ 
   \hline
\end{tabular}
\caption{Posterior mean estimates of cycle offset resolution, precision and housekeeping normalisation parameters for single-target models based on each target (values in brackets = 95\% credible intervals). \label{tab:single_target_offset_models_summary}} 
\end{table}


% \begin{table}[!t]
% \centering
%  {\begin{tabular}{|l|lll|}
% \hline
% Target &  $\lambda_C$ & $\sigma_C$ & $\rho_C$  \\
% \hline
% ARG1 &  -0.77 (-0.84,  -0.71) & 0.61 (0.50,  0.74) & 1.18 (1.05,  1.30) \\ 
% BATF  &  -1.14 (-1.31,  -0.97) & 0.68 (0.59,  0.77) & 0.88 (0.78,  0.99) \\ 
% C3AR1  &  -0.87 (-0.97,  -0.77) & 0.56 (0.46,  0.68) & 0.59 (0.49,  0.69) \\ 
% C9orf95 &  -0.74 (-0.99,  -0.49) & 0.85 (0.74,  0.98) & 0.55 (0.42,  0.68) \\  
% CD163 &  -0.62 (-0.70,  -0.55) & 0.43 (0.35,  0.50) & 0.40 (0.33,  0.47) \\ 
% CEACAM1 &  -0.68 (-0.78,  -0.59) & 0.88 (0.74,  1.01) & 0.54 (0.40, 0.68) \\ 
% CTSL1 &  -1.94 (-2.16,  -1.71) & 1.30 (1.07,  1.53) & 1.14 (0.95,  1.34) \\ 
% GADD45A &  -0.75 (-0.90,  -0.61) & 0.96 (0.75,  1.14) & 0.32 (0.17,  0.47) \\ 
% GNA15 &  -0.82 (-1.02,  -0.62) & 0.62 (0.54,  0.71) & 0.97 (0.87,  1.06) \\ 
% HK3 &  -0.74 (-0.81,  -0.66) & 0.41 (0.36,  0.46) & 0.72 (0.66, 0.78) \\ 
% HLA-DMB &  -0.81 (-1.36,  -0.25) & 3.13 (2.79,  3.53) & 0.73 (0.37, 1.08) \\ 
% IFI27 &  -0.94 (-1.05, -0.83) & 1.80 (1.51,  2.12) & 0.66 (0.42,  0.90) \\ 
% ISG15 &  -0.82 (-0.95,  -0.68) & 1.62 (1.37,  1.90) & 0.91 (0.69, 1.14) \\ 
% JUP &  -1.49 (-1.73,  -1.24) & 2.36 (2.04,  2.69) & 1.50 (1.20,  1.81) \\ 
% KCNJ2 &  -0.77 (-0.90,  -0.65) & 0.52 (0.41,  0.66) & 0.14 (0.05, 0.24) \\ 
% KIAA1370 &  -0.25 (-0.50,  -0.00) & 0.74 (0.60,  0.89) & 0.45 (0.32,  0.57) \\ 
% LY86 &  -0.82 (-0.98,  -0.66) & 0.55 (0.45,  0.64) & 0.88 (0.80, 0.97) \\ 
% OASL &  -0.89 (-0.96,  -0.81) & 0.65 (0.53,  0.78) & 0.87 (0.75, 0.99) \\ 
% OLFM4 &  -1.07 (-1.19,  -0.95) & 1.50 (1.31,  1.71) & 0.56 (0.35,  0.78) \\ 
% PDE4B &  -0.90 (-1.26,  -0.53) & 1.32 (1.17,  1.49) & 0.99 (0.80,  1.18) \\ 
% PER1 &  -1.41 (-1.78,  -1.05) & 2.07 (1.72,  2.45) & 0.61 (0.32,  0.89) \\ 
% PSMB9 &  -0.88 (-1.02,  -0.74) & 0.51 (0.41,  0.62) & 0.86 (0.77,  0.94) \\ 
% RAPGEF1 &  -0.88 (-1.36,  -0.40) & 1.26 (1.02,  1.51) & 0.97 (0.75,  1.17) \\ 
% TGFBI &  -0.66 (-0.75,  -0.58) & 0.53 (0.45,  0.62) & 0.85 (0.76,  0.94) \\ 
% ZDHHC19 &  -2.58 (-2.95,  -2.25) & 2.85 (2.10,  3.70) & 1.32 (0.96,  1.69) \\ 
% \hline
% \end{tabular}}
% \caption{Posterior mean estimates of cycle offset resolution, precision and housekeeping normalisation parameters for single-target models based on each target (values in brackets = 95\% credible intervals).\label{tab:single_target_rate_models_summary}}
% \end{table}

\begin{table}[ht]
\centering
\begin{tabular}{cccc}
  \hline
{\textbf{Target}} & $\bm{\lambda_B}$ & $\bm{\sigma_B}$ & $\bm{\rho_B}$ \\ 
  \hline
ARG1 & 0.02 (0.02, 0.02) & 0.03 (0.03, 0.03) & -0.03 (-0.03, -0.03) \\ 
  BATF & 0.03 (0.02, 0.03) & 0.03 (0.03, 0.03) & -0.02 (-0.02, -0.01) \\ 
  C3AR1 & 0.02 (0.02, 0.02) & 0.02 (0.02, 0.03) & -0.01 (-0.02, -0.01) \\ 
  C9orf95 & 0.02 (0.01, 0.02) & 0.03 (0.02, 0.03) & -0.01 (-0.02, -0.01) \\ 
  CD163 & 0.00 (0.00, 0.01) & 0.03 (0.02, 0.03) & -0.01 (-0.01, -0.01) \\ 
  CEACAM1 & 0.00 (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.01 (-0.02, -0.01) \\ 
  CTSL1 & 0.03 (0.02, 0.03) & 0.03 (0.03, 0.03) & -0.01 (-0.02, -0.01) \\ 
  GADD45A & 0.02 (0.01, 0.02) & 0.03 (0.03, 0.03) & -0.02 (-0.02, -0.01) \\ 
  GNA15 & 0.02 (0.01, 0.02) & 0.02 (0.02, 0.02) & -0.02 (-0.02, -0.02) \\ 
  HK3 & 0.00 (0.00, 0.00) & 0.02 (0.02, 0.02) & -0.01 (-0.02, -0.01) \\ 
  HLA-DMB & 0.01 (0.00, 0.01) & 0.03 (0.03, 0.03) & -0.01 (-0.01, -0.01) \\ 
  IFI27 & 0.01 (0.00, 0.01) & 0.03 (0.02, 0.03) & -0.01 (-0.02, -0.01) \\ 
  ISG15 & 0.01 (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.02 (-0.02, -0.01) \\ 
  JUP & 0.02 (0.02, 0.02) & 0.03 (0.03, 0.03) & -0.02 (-0.02, -0.01) \\ 
  KCNJ2 & 0.00 (0.00, 0.00) & 0.02 (0.02, 0.02) & -0.01 (-0.01,  0.00) \\ 
  KIAA1370 & 0.02 (0.01, 0.03) & 0.04 (0.03, 0.04) & -0.01 (-0.02, -0.01) \\ 
  LY86 & 0.02 (0.01, 0.02) & 0.02 (0.02, 0.02) & -0.02 (-0.02, -0.01) \\ 
  OASL & 0.01 (0.01, 0.02) & 0.02 (0.02, 0.03) & -0.02 (-0.02, -0.01) \\ 
  OLFM4 & 0.01 (0.01, 0.01) & 0.03 (0.03, 0.03) & -0.01 (-0.01, -0.01) \\ 
  PDE4B & 0.01 (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.01 (-0.02, -0.01) \\ 
  PER1 & 0.04 (0.03, 0.05) & 0.05 (0.05, 0.06) & -0.01 (-0.02, -0.01) \\ 
  PSMB9 & 0.01 (0.00, 0.01) & 0.02 (0.01, 0.02) & -0.01 (-0.02, -0.01) \\ 
  RAPGEF1 & 0.00 (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.02 (-0.02, -0.02) \\ 
  TGFBI & 0.01 (0.00, 0.01) & 0.02 (0.02, 0.03) & -0.01 (-0.02, -0.01) \\ 
  ZDHHC19 & 0.04 (0.04, 0.04) & 0.05 (0.04, 0.05) & -0.02 (-0.02, -0.01) \\ 
   \hline
\end{tabular}
\caption{Posterior mean estimates of rate resolution, precision and housekeeping normalisation parameters for single-target models based on each target (values in brackets = 95\% credible intervals). \label{tab:single_target_rate_models_summary}} 
\end{table}

% \begin{table}[!t]
% \centering
%  {\begin{tabular}{|l|lll|}
% \hline
% Target &  $\lambda_B$ & $\sigma_B$ & $\rho_B$  \\
% \hline
% ARG1 & 0.02 (0.02, 0.02) & 0.03 (0.02, 0.03) & -0.03 (-0.03, -0.02) \\
% BATF & 0.03 (0.02, 0.03) & 0.02 (0.02, 0.03) & -0.02 (-0.02, -0.01) \\
% C3AR1 & 0.02 (0.02, 0.02) & 0.02 (0.02, 0.02) & -0.01 (-0.02, -0.01) \\ 
% C9orf95 & 0.02 (0.01, 0.03) & 0.02 (0.02, 0.03) & -0.01 (-0.02, -0.01) \\ 
% CD163 & 0.00 (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.01 (-0.01, -0.01) \\ 
% CEACAM1 & 0.00 (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.01 (-0.02, -0.01) \\ 
% CTSL1 & 0.03 (0.02, 0.03) & 0.03 (0.02, 0.03) & -0.01 (-0.02, -0.01) \\ 
% GADD45A & 0.02 (0.01, 0.02) & 0.02 (0.02, 0.03) & -0.02 (-0.02, -0.01) \\ 
% GNA15 & 0.01 (0.00, 0.02) & 0.02 (0.01, 0.02) & -0.02 (-0.02, -0.01) \\ 
% HK3 & 0.00  (0.00, 0.00) & 0.02 (0.02, 0.02) & -0.01 (-0.02, -0.01) \\ 
% HLA-DMB & 0.01 (0.00, 0.01) & 0.03 (0.03, 0.03) & -0.01 (-0.01, -0.01) \\ 
% IFI27 & 0.01 (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.01 (-0.01, 0.00) \\ 
% ISG15 & 0.00  (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.01 (-0.02, -0.01) \\ 
% JUP & 0.02 (0.02, 0.02) & 0.03 (0.02, 0.03) & -0.02 (-0.02, -0.01) \\ 
% KCNJ2 & 0.00  (0.00, 0.00) & 0.01 (0.01, 0.02) & -0.01 (-0.01, 0.00) \\ 
% KIAA1370 & 0.01 (0.00, 0.02) & 0.03 (0.03, 0.04) & -0.01 (-0.02, -0.01) \\ 
% LY86 & 0.02 (0.01, 0.02) & 0.02 (0.01, 0.02) & -0.02 (-0.02, -0.01) \\ 
% OASL & 0.01 (0.01, 0.02) & 0.02 (0.02, 0.02) & -0.02 (-0.02, -0.01) \\ 
% OLFM4 & 0.01 (0.01, 0.02) & 0.03 (0.03, 0.04) & -0.01 (-0.01, 0.00) \\ 
% PDE4B & 0.01 (0.00, 0.02) & 0.02 (0.02, 0,02) & -0.01 (-0.02, -0.01) \\ 
% PER1 & 0.04 (0.03, 0.05) & 0.05 (0.04, 0.06) & -0.01 (-0.02, -0.01) \\ 
% PSMB9 & 0.01 (0.00, 0.01) & 0.01 (0.01, 0.02) & -0.01 (-0.02, -0.01) \\ 
% RAPGEF1 & 0.00  (0.00, 0.01) & 0.02 (0.02, 0.02) & -0.02 (-0.02, -0.01)\\ 
% TGFBI & 0.01 (0.00, 0.01) & 0.02 (0.01, 0.02) & -0.01 (-0.02, -0.01) \\ 
% ZDHHC19 & 0.04 (0.04, 0.05) & 0.04 (0.04, 0.05) & -0.02 (-0.02, -0.01) \\ 
% \hline
% \end{tabular}}
% \caption{Posterior mean estimates of rate resolution, precision and housekeeping normalisation parameters for single-target models based on each target (values in brackets = 95\% credible intervals).\label{tab:single_target_rate_models_summary}}
% \end{table}

To further elucidate the range of behaviours captured by these single-target models, we simulated from the posterior predictive distributions of $\hat{B}$ and $\hat{C}$ for four targets ({CEACAM1}, {BATF}, {PER1}, and {ZDHHC19}), to parameterise scaled Gompertz curves. We used these samples to simulate scaled Gompertz curves from targets showing different types of association between \glsxtrshort{rna} input and amplification rate/cycle offset. We simulated these curves at three \glsxtrlong{ns}-derived \glsxtrshort{rna} input values (results shown in Figure~\ref{fig:sm_comp_plot}). CEACAM1 displays negligible rate dependence on \glsxtrshort{rna} input, whereas all three others appear qualitatively different in shape across the gene expression range. \emph{PER1} and \emph{ZDHHC19} exhibit the greatest offset resolution, but this is accompanied by substantially worse precision. A key point to emphasise is that, unlike in the case of \gls{qpcr}, opting for an approach based purely on time-to-threshold will under-utilise the full information content of \gls{qrtlamp} curves. As rates of amplification also appear important for characterising \gls{qrtlamp} reactions, parameter estimates of the rates could be used in tandem with cycle offsets for target quantitation.

\begin{figure}[!tpb] 
\centering
\includegraphics[width=6in]{../figures/chapter2/single_target_model_comparisons.png}
\caption{Simulated amplification curves for four primers ({CEACAM1}, {RAPGEF1}, {BATF}, and {ZDHHC19}). Posterior predictive samples of Gompertz curves of scaled fluorescence based on posterior predictive samples from models~\ref{eq:bsm} and~\ref{eq:csm}. For each target, curves are simulated at three different \glsxtrshort{rna} input (\gls{ns}) levels -- note that higher \gls{ns} values correspond to higher expression and, generally, to lower cycle offsets (left-shifted amplification curves). Solid lines depict posterior means, while highlighted regions correspond to 95\% credible intervals. \label{fig:sm_comp_plot}}
\end{figure}


\subsection{Multi-target model identifies assay properties associated with LAMP resolution in patient data\label{sec:hierarchicalresults}}
We fitted the multi-target models described in Equations~\ref{eq:bhm}~and~\ref{eq:chm} to our clinical dataset using different combinations of primer/target features in the resolution function $\lambda(\bm{z}_{k}; \bm{\gamma})$. We considered the following forms for $\lambda$ in each analysis: \emph{a)} an intercept-only model $\lambda(\bm{\gamma}) = \gamma_0$; \emph{b)} for each of the three amplicon properties, a univariate linear function (e.g. for stem length $L_{\text{stem}}$, $\lambda(z_{k}; \bm{\gamma}) = \gamma_0 + \gamma_{L_{\text{stem}}} L_{\text{stem,k}}$); and \emph{c)} for each primer-specific quantity and primer pair, a bivariate model (e.g. in the case of free energy of annealing for the F1/B1 pair, $\lambda(\bm{z}_{k}; \bm{\gamma}) = \gamma_0 + \gamma_{\Delta G_a^{\text{F1}}} \Delta G_{a,k}^{\text{F1}} + \gamma_{\Delta G_a^{\text{B1}}} \Delta G_{a,k}^{\text{B1}}$). Note again that in this setting we only have one candidate primer set for each target and so no dependence on primer set $l$.

Posterior summaries of the resolution function coefficients as well as the \gls{loopsis} stacking weights for each of these models appear in Figure~\ref{fig:hierarchical_stacking_b} for models based on estimated rate and in Figure~\ref{fig:hierarchical_stacking_c} for those based on estimated cycle offsets. In each case, for the stacking weights (left-hand panels of Figures~\ref{fig:hierarchical_stacking_b}/\ref{fig:hierarchical_stacking_c}), the higher the stacking weight, the higher quality predictions the model was judged to have made for a left-out sample in comparison to alternative models. In the right-hand panel of Figures~\ref{fig:hierarchical_stacking_b}/\ref{fig:hierarchical_stacking_c}, we show posterior summaries of effect sizes (i.e. $\bm{\gamma}_B$ and $\bm{\gamma}_C$ coefficients) associated with each of the primer/target properties used as features in the resolution function, $\lambda$. 

\begin{figure}[!tpb] 
\centering
\includegraphics[width=5in]{../figures/chapter2/bhm_eval_fig.png}
\caption{Posterior summaries of multi-target models for rate resolution specified in Sections~\ref{sec:hierarchicalmodel}~and~\ref{sec:hierarchicalresults}, based on combinations of primer/target properties, with their associated Bayesian model stacking weights (left) and $\bm{\gamma}_B$-coefficient effects sizes (right). Effect sizes are posterior means, with error bars delineating 95\% credible intervals. Notation as given in Table~\ref{tab:properties}.\label{fig:hierarchical_stacking_b}}
\end{figure}

\begin{figure}[!tpb] 
\centering
\includegraphics[width=5in]{../figures/chapter2/paper_eval_fig.png}
\caption{Posterior summaries of multi-target models for cycle offset resolution specified in Sections~\ref{sec:hierarchicalmodel}~and~\ref{sec:hierarchicalresults}, based on combinations of primer/target properties, with their associated Bayesian model stacking weights (left) and $\bm{\gamma}_C$-coefficient effects sizes (right). Effect sizes are posterior means, with error bars delineating 95\% credible intervals. Notation as given in Table~\ref{tab:properties}.\label{fig:hierarchical_stacking_c}}
\end{figure}

We found quantities associated with cycle offset resolution in both positive and negative directions whose corresponding models were assigned strong weightings by the \gls{loopsis} procedure. Increased stem length, complexity in the F3/B3 primer pair, free energy of annealing in F2/B2 regions of the FIP/BIP primers, and free energy of secondary structure formation in F3/B3 were associated with improved resolution (negative effect sizes), and all corresponding models received relatively high stacking weights.
Immediately striking is the observation that, despite the widespread assumption that design of the FIP/BIP primers is most influential in determining \gls{lamp} performance, we detect surprisingly strong associations with properties of other primer pairs. This is a strong signal suggesting the need for future follow-up work.
These associations, particularly for F3, may reflect the importance of the primer to the synthesis of a full-length template, resulting in more sites for primer annealing and amplification (Figure~\ref{fig:lampoverview}). Increases in complexity of LF/LB primers, free energy of self-dimerisation for FIP/BIP, and free energy of secondary structure formation in LF/LB were associated with worse resolution (positive effect sizes), with corresponding models also receiving high stacking weights. We do observe some primer pairs for which posterior mean effect sizes were in opposing directions. For example, free energy of self-dimerisation shows a positive effect for the B3 primer and a negative effect for F3. We see that no such models were assigned large stacking weights. We might expect these different associations with resolution due to stochasticity of the \gls{lamp} assay in the production of intermediates from either strand of the target.

We also observe associations between primer/amplicon properties and rate resolution (Figure~\ref{fig:hierarchical_stacking_b}). Note that in this case a positive effect size corresponds to improved resolution. Some properties appear to be associated with rate resolution in the same way as offset resolution: for example, 4-complexity of F3/B3 are associated with improved resolution in each case. Others, such as stem length, are associated with worsened rate resolution but improved offset resolution. Others still show strong effects in one model but not in another, including free energy of self-dimerisation of the F3/B3 pair. Unlike for offset models, one of the most strongly identified contributors to rate resolution (free energy of annealing of the F1/B1 regions) exhibits distinct reversed directions of effect between F1 and B1 regions.

\subsection{Varying qRT-LAMP primer sequences identifies distinct across-target and within-target patterns of assay resolution \label{sec:ivt_results}}
In this section we describe our use of a distinct, novel synthetic dataset (preparation described in Section~\ref{sec:data} and \citealp{remmel_diagnostic_2022}) to address with more granularity the direct effects of changes to primer sequences on assay resolution. While in Section~\ref{sec:hierarchicalresults} we identified interesting and unexpected associations with other primer pairs, here we focused on FIP and BIP primers -- future work will address comparable questions for F3/B3/LF/LB. We also focus here on offset resolution only. In the previous analysis, we pooled our data across targets to estimate associations of assay properties with resolution. However, while we showed associations of these properties with resolution across targets, we could not determine for any \emph{individual} target whether variation in these properties was associated with variation in resolution. To address the latter objective (arguably more important for \gls{lamp} optimisation), we generated a \gls{ivt} \glsxtrshort{rna} dataset comprising multiple primer sets (containing multiple combinations of the FIP and BIP primers) for each target. We fitted models with two different forms of resolution function. In the first form, for each combination of assay properties we fitted a model based on those properties alone with no target-level grouping terms (\emph{across-target} models). In this way, the resolution function $\lambda$ did not depend on target ($k$) other than through the primer set ($l$) features $\bm{z}_{kl}$ (e.g., $\lambda(z_{kl}; \bm{\gamma}) = \gamma_0 + \gamma_{L_{\text{stem}}} L_{\text{stem},k,l}$).
 
In the second model form (\emph{within-target}), we specified a hierarchical random effects term (varying by target) for the resolution intercept, $\gamma_0$. In this way, we were able to infer the dependence of resolution on primer set properties while accounting for intrinsic target-to-target variation in resolution. Summarised inferences and model stacking weights after fitting of the across-target and within-target models to the \gls{ivt} \glsxtrshort{rna} data appear in Figure~\ref{fig:ivt_eval}. Note again that these models are not a subset of those shown in Figure~\ref{fig:hierarchical_stacking_c}, but rather similar models applied to a novel synthetic \gls{ivt} dataset with systematically varied FIP and BIP primers.

\begin{figure}[!tpb] 
\centering
\includegraphics[width=6in]{../figures/chapter2/paper_ivt_eval_fig.png}
\caption{Posterior summaries of multi-target models as specified in Sections~\ref{sec:hierarchicalmodel}~and~\ref{sec:ivt_results} applied to \gls{ivt} \glsxtrshort{rna} data. Effect sizes for $\bm{\gamma}$-coefficients from across-target (center left) and within-target (center right) models, as well as associated model stacking weights (far left/right respectively). Error bars delineate 95\% credible intervals. \label{fig:ivt_eval}}
\end{figure}

We note that the across-target models gave far more inferences than the within-target models of high-confidence, high-magnitude effect sizes of assay properties on resolution. Relatively few properties were inferred to have effect sizes of high magnitude (many 95\% credible intervals contain 0) for within-target models. Importantly, these results indicate that much of the variation in resolution explained in previous models may be due to intrinsic target-to-target variation not attributable to the assay properties we considered. Of the within-target models, those based on complexity of the FIP primer, free energy of self-dimerisation of the FIP primer, and free energy of secondary structure formation of the FIP primer had significant posterior mean effect sizes (i.e. 95\% posterior credible intervals not containing zero). Notably, several of these within-target inferences differ in directionality from their across-target counterparts, indicating that not accounting for intrinsic target-to-target variation could be masking the nature of the association between assay properties and resolution. Furthermore, these results highlight the importance of utilising multiple primer sets per target in characterising this variation. Interestingly, we do note that both within- and across-target models agreed in allocation of the largest stacking weight to free energy of secondary structure formation for FIP/BIP, a novel quantity for consideration in the design of \gls{qrtlamp} assays.


\section{Discussion \label{sec:discussion}}
We have presented and applied a novel model-based framework to characterise quantitative performance of the \gls{qrtlamp} assay. We first introduced a non-linear least-squares fitting procedure based on Gompertz curves that allows for fast, accurate characterisation of \gls{qrtlamp} reactions. Based on LAMP reaction curve characteristics derived from this Gompertz curve-based fitting procedure, our Bayesian hierarchical models  have generated new insights into the \gls{qrtlamp} reaction and its properties at the single- and multi-target levels. At the single-target level, our analysis identified \glsxtrshort{rna}-independent and \glsxtrshort{rna}-dependent associations with qualitative characteristics and quantitative performance of \gls{qrtlamp} in a dataset of patients with suspected acute infection. Our investigation of the effects of various sequence-based and thermodynamic primer/target properties on \gls{qrtlamp} performance identified novel associations between some of these properties and \gls{lamp} assay resolution. We've not only adapted our framework to single-target and multi-target settings but also to different \gls{qrtlamp} data modalities (from \gls{ivt} \glsxtrshort{rna} and clinical samples). Finally, we have used our \gls{ivt} \glsxtrshort{rna} data and modelling framework to investigate the extent to which associations observed between primer/target properties and \gls{lamp} performance will be borne out by artificially altering primers as part of an assay optimisation workflow. 

We highlight several advantages of our approach. The single-target models introduced in Section~\ref{sec:singletargetmodel} and applied in Section~\ref{sec:sensitivityprecision} provide a formal framework for inference with uncertainty of \gls{qrtlamp} performance measures such as resolution and precision. This approach allows assay developers to ascertain the quantitative performance of an predesigned \gls{lamp} target. Such inferences may in turn motivate optimisation of the given target or, in light of poor target performance, alternative target selection as part of a broader panel of target biomarkers. Furthermore, both our single-target and multi-target models allow decomposition of variation in \gls{lamp} reaction characteristics into components useful for \gls{mrna} quantitation (i.e. amplification rate and cycle offset). By incorporating different features of the assay into model components for \glsxtrshort{rna}-dependent and \glsxtrshort{rna}-independent variation, we can uncover potentially tunable determinants of \gls{lamp} assay performance. Indeed, our framework applied to \gls{ivt} \glsxtrshort{rna} data revealed significant associations between primer and target features and assay resolution. While \gls{ivt} \glsxtrshort{rna} data are useful for assay development due to their cost and ease of handling, they are not biologically representative and would likely not recapitulate assay performance seen on clinical samples. As we could adapt our models to both \gls{ivt} \glsxtrshort{rna} and clinical samples, we were able to compare our inferences across different data types and to better anticipate clinical performance of a given assay. We see our framework and analysis as important first steps towards a system for \gls{lamp} assay design and optimisation.

However, the results of our hierarchical modelling (Section~\ref{sec:ivt_results}) highlight the complexities of optimising \gls{qrtlamp} assay performance. While FIP and BIP primers are known to play a central role in LAMP-based amplification, our model-based analysis (Figures~\ref{fig:hierarchical_stacking_b} and~\ref{fig:hierarchical_stacking_c}) provides evidence that other primer characteristics may also affect assay performance. Models based on loop amplification and inner primer pairs were identified by model stacking as producing some of the most robust and predictive associations. Furthermore, our findings suggest that increasing a given property (for example, free energy of self-dimerisation) may be associated with improved resolution for one primer pair (FIP/BIP) but the opposite for another (LF/LB). As changing one oligo property (e.g. secondary structure formation) may alter another (propensity to self-dimerise), our results support joint consideration of all primer pair characteristics as part of assay optimisation efforts. %In addition, we highlight the importance of model validation, provided here by \gls{loopsis} approximations to leave-one-out cross-validation. High stacking weight scores indicated good agreement between clinical and (within-target) \gls{ivt} models. 
We also observe, for several properties, a difference in 
magnitude % directionality 
of effect sizes between forward and backward primers (Figure~\ref{fig:ivt_eval}). Subject to further investigation, we hypothesise that this difference in 
association strength % directionality 
between members of a primer pair could arise due to the orientation of initial \glsxtrshort{rna} inputs to the \gls{lamp} assay. For example, B3 primers could anneal to \glsxtrshort{rna} inputs before reverse transcription while F3 primers -- relying on the generation of an F3c region -- could not anneal until after reverse transcription, and this could explain some degree of within-primer pair disparity. However, this trend is far from clear and consistent across different quantities and primer pairs, so much further investigation is required.

Further work is required to accurately predict performance in a proposed assay for a potentially novel target. As part of our efforts towards a more predictive model, we analyzed multiple primer sets per target with our \gls{ivt} \glsxtrshort{rna} dataset. Indeed, we found for multiple models that the effects of varying properties of primers and amplicons differ in their strength and direction when viewed across-target vs within-target (Figure~\ref{fig:ivt_eval}). Put another way, accounting for target-to-target differences in resolution not due to the assay features attenuated the strength of association between the features and resolution. These results suggest that other assay features may be explaining target-to-target differences in resolution and that predictive models based on pooling all targets together (i.e. across-target models) may not generalise well to new targets. Interestingly, recent work on predictive modelling of \gls{pcr} amplification from primer/target features showed good performance in held-out test sets~\citep{doring_modeling_2019}. This performance could be due to both a focus on a highly related set of targets which might be expected to share assay performance properties as well as due to random data subsampling to create training, validation and test data splits for model development. In this setting, all splits would likely include observations for a given target (a kind of information leakage), complicating interpretation of the model's generalisability. However, if we consider a highly related family of targets to be similar to a single target, the results of~\citet{doring_modeling_2019} are consistent with our findings that within-target trends (rather than general trends shared across targets) in the relationship between \gls{lamp} assay performance and primer properties are important for optimisation of a given target or target family. Reliably determining the optimal properties of a primer set to achieve a pre-specified level of quantitative performance would allow for less trial-and-error in the development process, and ongoing and future work will be required to add both more targets and primer sets to our \gls{ivt} \glsxtrshort{rna} dataset.
%From these distinct effects we emphasise that studies based purely on across-target observational data may be misleading, or at least fail to capture the whole picture.
%That said, knowledge of across-target differences in \gls{lamp} assay performance from target to target can be useful for target identification: if one is able to predict reliably whether a given target is likely to have good resolution, then significant development time and costs could be saved. 
%If we consider the, we note that robust .  This distinction is important, since within-target effects can only be identified with data that contains multiple candidate primer sets per target.

To extend and more comprehensively validate the results presented here, a follow-up study will be required to profile many more targets with -- crucially -- multiple combinations of primers for each target. In particular, we note that while our \gls{ivt} \glsxtrshort{rna} data analysis included multiple primer sets per target, we were only able to assay three targets due to time/cost constraints. Further data collection will also permit investigation of additional candidate assay properties, including \gls{gdna} interference, sensitivity to reagents and other sequence-based/thermodynamic quantities. In addition, while \gls{ivt} \glsxtrshort{rna} data are useful for assay development due to their cost and ease of handling, they are not biologically representative. The degree to which conclusions drawn from \gls{ivt} \glsxtrshort{rna} translate to  clinical samples is the subject of ongoing and future investigation. Also, while in our multi-target modelling work we have investigated variation in resolution with respect to cycle offset, we have not provided a complementary analysis for the other properties shown to be important in our single-target modelling section. These include resolution with respect to amplification rate, and precision. Extending our analysis to identify associations between these performance characteristics and assay properties is the subject of ongoing and future work. In addition, for simplicity, we assumed a linear relationship between assay properties and resolution. Further investigations will involve nonlinear relationships between these quantities.


% \subsection{Integration into assay optimisation workflows \label{sec:integration}}
% Experimental approaches to primer optimisation are extremely labour-intensive and a major financial and temporal bottleneck in \gls{lamp} assay development. When designing \gls{lamp}-based tests for a given application, the standard approach is to identify a target or set of targets of biological significance using conventional genomics technologies, and then to design an assay to rapidly and conveniently measure this signature \citep{gliddon_genome-wide_2018}. This two-step process is necessary since once a gene target or set of targets has been identified as clinically relevant, the amount of investment required to convert these targets into a functional test means that there is little scope to refine the initial target set on the basis of amenability to measurement with \gls{lamp}. However, this can lead to seriously under-optimised tests, since the performance of \gls{lamp} was not taken into account when determining the most relevant targets \citep{pennisi_translation_2021}. We envisage an alternate workflow, in which assay optimisation is conducted in parallel with target selection. By intelligently utilising \gls{qrtlamp} data within a modelling framework to guide primer optimisation, and conversely utilising model outputs to inform target selection, practitioners may be able to produce better performing point-of-care tests more quickly and cheaply.

% More immediately, software designed to aid \gls{lamp} primer design may benefit from being extended to include optimisation quantities such as those analysed here, particularly in the context of quantitative \gls{lamp}. Of common workflows currently used, PrimerExplorer \citep{eiken_primerexplorer_2019} includes some functionality to avoid secondary structure, LAVA \citep{torres_lava_2011} incorporates melting temperature information, and GLAPD \citep{jia_glapd_2019} allows the user to check for the possibility of inter-primer bonding, yet no individual piece of software yet achieves comprehensive prediction of \gls{lamp} performance on the basis of the range of primer properties available (some of which are presented here). Again, this gap is even more strongly evident for \gls{qrtlamp}, where the interacting effects of different primer/targets properties may be more nuanced.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%     please remove the " % " symbol from \centerline{\includegraphics{fig01.eps}}
%     as it may ignore the figures.
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%






\section{Conclusion}

This work provides an extensible platform to address the significant knowledge gap in enabling reliable quantitative design and use of the \gls{qrtlamp} assay. 
%While some primer design frameworks include functionality to account for desirable or undesirable primer set features, such as by avoiding secondary structure formation~\citep{eiken_primerexplorer_2019} or primer dimerisation~\citep{jia_glapd_2019}, the field is as of yet a long way from a holistic picture of the landscape of \gls{lamp} assay optimisation and quantitation. 
Our analysis has uncovered important insights of \glsxtrshort{rna}-dependent and \glsxtrshort{rna}-independent effects on \gls{qrtlamp} assay behaviour as well as assay properties associated with improved quantitative resolution of \gls{lamp}. Our efforts have also raised important considerations for constructing truly predictive systems of assay performance. With further data collection, expansion of our set of assay features, and extension of our modelling framework, we hope to take another important step towards principled design of robust, clinically performant \gls{qrtlamp} assays. 





\dobib % renders bibliography (only when compiling for chapter)


\end{document}
