\documentclass[../thesis.tex]{subfiles}

\begin{document}

\section{Cancer as a disease of the genome}
Cancer does not require much introduction even to the lay reader; direct or indirect experience of cancer is universal. It is consistently ranked amongst the leading causes of global mortality, and multiple subtypes of cancer are projected to increase in their ranking and share of worldwide premature deaths over the coming decades \citep{mathers_projections_2006}, including in the developing world \citep{kanavos_rising_2006}. Beyond its direct death toll, cancer is responsible for the expenditure of trillions of dollars per year in cost of care and lost economic output \citep{wild_world_2020}. While huge gains in the understanding, prevention and treatment of cancer have been made in recent years, many challenges remains in scalably advancing each of these three categories. Modern cancer treatments in particular are often extremely expensive, with drug development costs increasing and consequently inflating the price of access to therapeutics \citep{howard_pricing_2015}. In short, there is much to be hopeful about in oncology, but it is by no means guaranteed that the current revolutions being enjoyed in scientific understanding will translate fully to equitable clinical benefit.

In order to understand the state of play in cancer research, we need to know a little about the nature of cancer itself, and a little about the nature of modern molecular biology. A key starting point is that cancer is not a unitary disease;  two patientsâ€™ tumours may be caused by different processes, occur in different tissues, and have very different molecular and physiological effects \citep{wittekind_tnm_2016}. More so than any other disease, every cancer is unique. It is therefore natural to ask what unifying features of all cancers justify their joint classification. The modern answer is distinct from the historical answer. Before the advent of genetics, cancers of disparate tissues of origin were grouped together under the unifying observation of malignant growths crossing over physiological boundaries. Towards the end of the 19th century, it was recognised that aberrant patterns of cell reproduction were a common feature of cancers \citep{weinstein_history_2008}. By the early 1900s, with the writings of scientists such as Theodor Boveri \citep[see][for a modern translation]{boveri_concerning_2008}, an answer would be formulated foreshadowing our current understanding, although it wouldn't be until far later that this explanation was fully accepted. Boveri proposed that 'chromosomal abnormalities' gave rise to the conversion of normal cells to malignant neoplasms. In modern nomenclature, the chromosomal abnormalities to which he referred would be regarded as (a specific kind of) mutations. It is these that lay the groundwork for uncontrolled cellular reproduction and all the other associated hallmarks of cancer\footnote{While some rare cancers such as \acrfull{pv} may involve uncontrolled production of cells that do not themselves harbour mutations (in this case, mature red blood cells do not contain DNA at all), this is still the downstream effect of mutations in other cell types. For example, in \acrshort{pv} this is most commonly a mutation of the \textit{JAK2} gene in hematopoietic stem cells \citep{tefferi_jak2_2007}.}, such as avoiding detection from the body's defenses (immunosuppression/evasion), recruiting a local blood supply (angiogenesis), and invasion of separate tissues (metastasis) \citep{hanahan_hallmarks_2011}. Crucially, mutations convert previously normal or benign cell populations into tumours. The mutations that were observable via optical microscopy to scientists in the first half of the twentieth century were structural mutations involving large-scale chromosomal translocations or deletions. It wasn't until the identification of the structure of DNA and its role as the primary mechanism of inheritance by Franklin, \citet{watson_molecular_1953} and the subsequent development of molecular genetics that the discrete nature of biological information was fully appreciated. Later developments in DNA sequencing, beginning with the work of \citet{sanger_dna_1977} allowed a fuller understanding of mutations as changes to the sequence of nucleotide bases that constitutes DNA. The science of cancer continued to progress by associating DNA mutations (errors of cellular information storage), with their mechanistic and functional consequences, in particular those that led to deregulation of normal cell-cycle control.

Now that we are armed with a general characterisation of cancer as the consequences of mutations in DNA leading to abnormal reproduction of cells, we can begin to appreciate the reasons for cancer's diversity. Since almost all cells in the body contain DNA and experience regular reproduction, cancer may occur in a wide range of tissues throughout the body\footnote{Note that tissues/cell types in which cancer is extremely uncommon tend to be those which experience very little reproduction, and so have little chance to accumulate mutations, e.g. neuronal cells and tissues making up the heart}. Furthermore, the size of the human genome (defined as the combined total of genetic information contained in DNA, comprising of around 20,000 genes and 3 billion nucleotide base pairs) means that, even with cancer being as common a disease as it is, simple statistical reasoning allows us to say with confidence that it is almost inconceivable that two given tumours would carry exactly the same constituent mutations (even without considering complicating factors such as tumour heterogeneity). This leads us to the modern era of molecular biology. Since the completion of the human genome project \citep{lander_initial_2001}, high-throughput sequencing, where large portions of the genome in their entirety are sequenced for a biological sample, has become ubiquitous and highly automated. We now have easy access to the precise locations of all mutations in the tumour genomes of many tens thousands of thousands of samples gathered across hundreds of studies via repositories such as \acrfull{tcga} \citep{weinstein_cancer_2013}. This gives us an opportunity to investigate a variety of fundamental questions with regards to the progression of cancer. One of these mirrors a classic debate of nature versus nurture in developmental biology. In this case, we wish to understand the extent to which the dynamics and trajectory of a tumour are pre-determined by the genetic damage it carries. We know that cancers are defined by their mutations, but a growing field of investigation is exploring the role of the environment in which a tumour finds itself in allowing it to flourish. For the remainder of this section, we will elaborate on the balance between these two views of the tumour genome.
\subsection{Nature: mutations in the driver's seat}
- Discussion of the ways in which mutations drive cancer, in particular oncogenes and tumour suppressors.
- Note that mutations can come from internal or external factors.
\subsection{Nurture: A warm (micro-) environment}
- Pull back on the importance of mutations, and look at the micro-environment, in particular hot and cold microenvironments. \citep{keenan_genomic_2019} \citep{boulter_fibrotic_2020}



\section{Genomic instability}
\subsection{The molecular mechanics of DNA damage}
\subsection{The hypermutated phenotype}
\subsection{Mismatch repair}
\section{Immunotherapy, checkpoint blockade and beyond}
Since the discovery of \gls{icb}\footnote{For their work on \gls{icb}, James Allison and Tasuku Honjo received the 2018 Nobel Prize for Physiology/Medicine \citep{ledford_cancer_2018}.}  \citep{ishida_induced_1992,leach_enhancement_1996},  there has been an explosion of interest in cancer therapies targeting immune response and \gls{icb} therapy is now widely used in clinical practice \citep{robert_decade_2020}.  \gls{icb} therapy works by targeting natural mechanisms (or \emph{checkpoints}) that disengage the immune system, for example the proteins \gls{ctla4} and \gls{pdl1} \citep{buchbinder_ctla-4_2016}. Inhibition of these checkpoints can promote a more aggressive anti-tumour immune response \citep{pardoll_blockade_2012}, and in some patients this leads to long-term remission \citep{gettinger_5-year_2019}. However, \gls{icb} therapy is not always effective \citep{nowicki_mechanisms_2018} and may have adverse side-effects, so determining which patients will benefit in advance of treatment is vital. 

\section{Clinical practice and pharmacoeconomics}

\subsection{Biomarkers as proxies of immunogenicity}
Discuss \acrshort{tmb} and \acrshort{msi}.

Exome-wide prognostic biomarkers for immunotherapy are now well-established -- in particular, \gls{tmb} is used to predict response to immunotherapy \citep{zhu_association_2019, cao_high_2019}.  \gls{tmb} is defined as the total number of non-synonymous mutations occurring throughout the tumour exome, and can be thought of as a proxy for how easily a tumour cell can be recognised as foreign by immune cells \citep{chan_development_2019}. However, the cost of measuring \gls{tmb} using \gls{wes} \citep{sboner_real_2011} currently prohibits its widespread use as standard-of-care.  Sequencing costs, both financial and in terms of the time taken for results to be returned, are especially problematic in situations where high-depth sequencing is required, such as when utilising blood-based \gls{ctdna} from liquid biopsy samples \citep{gandara_blood-based_2018}. The same issues are encountered when measuring more recently proposed biomarkers such as \gls{tib} \citep{wu_tumor_2019,turajlic_insertion-and-deletion-derived_2017}, which counts the number of frameshift insertion and deletion mutations. There is, therefore, demand for cost-effective approaches to estimate these biomarkers \citep{fancello_tumor_2019, golkaram_interplay_2020}.

\subsection{Liquid biopsy}
\citep{jensen_association_2020} \citep{genovese_clonal_2014} \citep{razavi_high-intensity_2019} \citep{schweizer_clinical_2019} \citep{annala_circulating_2018} \citep{goodall_circulating_2017}
\subsection{Targeted gene panels}

\section{Roadmap}

\dobib % renders bibliography (only when compiling for chapter only)


\end{document}